<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Pipeline 1. Data quality check • evoverse</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Pipeline 1. Data quality check">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">evoverse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/evoverse.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/0_packages.html">`evoverse` packages</a>
    </li>
    <li>
      <a href="../articles/1_qc_models.html">QC models inside `evoverse`</a>
    </li>
    <li>
      <a href="../articles/a1_Data_QC.html">Pipeline 1. Data quality check</a>
    </li>
    <li>
      <a href="../articles/a2_Chr_timing.html">Pipeline 2. Timing chromosome aneuploidy</a>
    </li>
    <li>
      <a href="../articles/a3_Deconv_vaf_karyo.html">Pipeline 3. Subclonal deconvolution by VAF and karyotypes</a>
    </li>
    <li>
      <a href="../articles/a4_Deconv_CCF.html">Pipeline 4. Subclonal deconvolution by CCF</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/evoverse">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Pipeline 1. Data quality check</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/evoverse/blob/master/vignettes/a1_Data_QC.Rmd"><code>vignettes/a1_Data_QC.Rmd</code></a></small>
      <div class="hidden name"><code>a1_Data_QC.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(evoverse)</a></code></pre></div>
<div id="pipeline-overview" class="section level1">
<h1 class="hasAnchor">
<a href="#pipeline-overview" class="anchor"></a>Pipeline overview</h1>
<p><img src="pipeline_qc.png" alt="drawing" height="100"></p>
<p><strong>Main function: </strong> `<code>pipeline_qc_copynumbercalls</code>.</p>
<div id="input-requirements" class="section level2">
<h2 class="hasAnchor">
<a href="#input-requirements" class="anchor"></a>Input requirements</h2>
<p>The following data are required to run this pipeline.</p>
<ul>
<li>one <a href="https://caravagn.github.io/CNAqc/">CNAqc</a> input dataset, which consists of:
<ul>
<li>somatic mutations (with read counts data), <code>mutations</code>;</li>
<li>absolute copy number segments, <code>cna</code>;</li>
<li>tumour purity, <code>purity</code>;</li>
<li>a genome reference supported by <a href="https://caravagn.github.io/CNAqc/">CNAqc</a>.</li>
</ul>
</li>
<li>pipeline parameters, which allow to:
<ul>
<li>identify a run by a <code>description</code>;</li>
<li>decide if input segments should be <a href="https://caravagn.github.io/CNAqcarticles/a5_smoothing.html">smoothed by CNAqc</a>, <code>smooth</code>;</li>
<li>decide how peaks should be <a href="https://caravagn.github.io/CNAqc/articles/a3_peaks_detection.html">matched by CNAqc</a>, <code>matching_epsilon_peaks</code>;</li>
<li>decide how CCF should be <a href="https://caravagn.github.io/CNAqc/articles/a4_ccf_computation.html">computed by CNAqc</a>, <code>CCF_computation</code>;</li>
</ul>
</li>
</ul>
</div>
<div id="pipeline-skeleton" class="section level2">
<h2 class="hasAnchor">
<a href="#pipeline-skeleton" class="anchor"></a>Pipeline skeleton</h2>
<p>The following steps are performed by <code>pipeline_qc_copynumbercalls</code>.</p>
<ol style="list-style-type: decimal">
<li><p>input mutations are mapped to copy number segments, integrating also tumour purity using <a href="https://caravagn.github.io/CNAqc/">CNAqc</a>; if required input segments are <a href="https://caravagn.github.io/CNAqcarticles/a5_smoothing.html">smoothed</a>.</p></li>
<li><p><a href="https://caravagn.github.io/CNAqc/articles/a3_peaks_detection.html">peak-detection</a> is used to determine the quality of each karyotype, with a QC <code>"PASS"</code> or <code>"FAIL"</code> assessment;</p></li>
<li><p><a href="https://caravagn.github.io/CNAqc/articles/a4_ccf_computation.html">CCFs are computed</a>, with a QC <code>"PASS"</code> or <code>"FAIL"</code> assessment per karyotype;</p></li>
<li><p><a href="https://caravagn.github.io/CNAqc/articles/a6_fragmentation.html">Arm-level fragmentation</a> is used to determine chromosome arms that seem to have an excess of small fragments;</p></li>
<li><p>On output table is computed to report QC results from all data.</p></li>
</ol>
</div>
</div>
<div id="example-run" class="section level1">
<h1 class="hasAnchor">
<a href="#example-run" class="anchor"></a>Example run</h1>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Use example data in the CNAqc package</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(example_dataset_CNAqc, <span class="dt">package =</span> <span class="st">'CNAqc'</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"></a>
<a class="sourceLine" id="cb2-4" data-line-number="4">pipeline =<span class="st"> </span><span class="kw"><a href="../reference/pipeline_qc_copynumbercalls.html">pipeline_qc_copynumbercalls</a></span>(</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">  <span class="dt">mutations =</span> CNAqc<span class="op">::</span>example_dataset_CNAqc<span class="op">$</span>snvs,  <span class="co"># Somatic mutations</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6">  <span class="dt">cna =</span> CNAqc<span class="op">::</span>example_dataset_CNAqc<span class="op">$</span>cna,         <span class="co"># Copy number segments</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">  <span class="dt">purity =</span> CNAqc<span class="op">::</span>example_dataset_CNAqc<span class="op">$</span>purity,   <span class="co"># Sample purity</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8">  <span class="dt">reference =</span> <span class="st">'GRCh38'</span>,                           <span class="co"># Genome reference</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9">  <span class="dt">description =</span> <span class="st">"Example dataset for QC"</span>,         <span class="co"># Dataset description</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10">  <span class="dt">smooth =</span> <span class="ot">TRUE</span>,                                  <span class="co"># Smooth copy number segments</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11">  <span class="dt">matching_epsilon_peaks =</span> <span class="fl">0.025</span>,                 <span class="co"># Peak-detection parameters</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12">  <span class="dt">ccf_method =</span> <span class="st">'ENTROPY'</span>                          <span class="co"># CCF computation</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13">  )</a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="co">#&gt;  [ Evoverse ~ Pipeline to QC somatic mutations and CNA segments ]</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="co">#&gt; ── Creating input object (smoothing TRUE) for sample Example dataset for QC ─────────────────────────────</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="co">#&gt; ── Preparing input for deconvolution ────────────────────────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-20" data-line-number="20"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb2-21" data-line-number="21"><span class="co">#&gt; ℹ Using reference genome coordinates for: GRCh38.</span></a>
<a class="sourceLine" id="cb2-22" data-line-number="22"><span class="co">#&gt; ℹ Drivers are annotated, but 'gene' column is missing, using mutation location.</span></a>
<a class="sourceLine" id="cb2-23" data-line-number="23"><span class="co">#&gt; ! Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span></a>
<a class="sourceLine" id="cb2-24" data-line-number="24"><span class="co">#&gt; ℹ Input n = 13141 mutations for 267 CNA segments (267 clonal, 0 subclonal)</span></a>
<a class="sourceLine" id="cb2-25" data-line-number="25"><span class="co">#&gt; x n = 178 mutations  cannot be mapped to segments and will be removed.</span></a>
<a class="sourceLine" id="cb2-26" data-line-number="26"><span class="co">#&gt; ✓ Mapped n = 12963 mutations to clonal segments (100% of input)</span></a>
<a class="sourceLine" id="cb2-27" data-line-number="27"><span class="co">#&gt; ── [ CNAqc ]  n = 12963 mutations in 267 segments (267 clonal + 0 subclonal). Genome reference: GRCh38. ─</span></a>
<a class="sourceLine" id="cb2-28" data-line-number="28"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-29" data-line-number="29"><span class="co">#&gt;    2:2  [n = 7478] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { chr19:15292495:15292496:G:A }</span></a>
<a class="sourceLine" id="cb2-30" data-line-number="30"><span class="co">#&gt;    4:2  [n = 1893] ■■■■■■■</span></a>
<a class="sourceLine" id="cb2-31" data-line-number="31"><span class="co">#&gt;    3:2  [n = 1625] ■■■■■■</span></a>
<a class="sourceLine" id="cb2-32" data-line-number="32"><span class="co">#&gt;    2:1  [n = 1563] ■■■■■■  { chr2:179431633:179431634:C:T }</span></a>
<a class="sourceLine" id="cb2-33" data-line-number="33"><span class="co">#&gt;    3:0  [n =  312] ■</span></a>
<a class="sourceLine" id="cb2-34" data-line-number="34"><span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 4.</span></a>
<a class="sourceLine" id="cb2-35" data-line-number="35"><span class="co">#&gt; ℹ Mutations annotated have 3 drivers.</span></a>
<a class="sourceLine" id="cb2-36" data-line-number="36"><span class="co">#&gt; ℹ Smoothing chr1: 37 segments.</span></a>
<a class="sourceLine" id="cb2-37" data-line-number="37"><span class="co">#&gt; [1-24] [27-30] [31-33] [34-37]</span></a>
<a class="sourceLine" id="cb2-38" data-line-number="38"><span class="co">#&gt; ℹ Smoothing chr10: 8 segments.</span></a>
<a class="sourceLine" id="cb2-39" data-line-number="39"><span class="co">#&gt; [1-2] [3-4] [5-8]</span></a>
<a class="sourceLine" id="cb2-40" data-line-number="40"><span class="co">#&gt; ℹ Smoothing chr11: 22 segments.</span></a>
<a class="sourceLine" id="cb2-41" data-line-number="41"><span class="co">#&gt; [1-7] [8-9] [10-22]</span></a>
<a class="sourceLine" id="cb2-42" data-line-number="42"><span class="co">#&gt; ℹ Smoothing chr12: 13 segments.</span></a>
<a class="sourceLine" id="cb2-43" data-line-number="43"><span class="co">#&gt; [2-3] [12-13]</span></a>
<a class="sourceLine" id="cb2-44" data-line-number="44"><span class="co">#&gt; ℹ Smoothing chr14: 2 segments.</span></a>
<a class="sourceLine" id="cb2-45" data-line-number="45"><span class="co">#&gt; [1-2]</span></a>
<a class="sourceLine" id="cb2-46" data-line-number="46"><span class="co">#&gt; ℹ Smoothing chr15: 9 segments.</span></a>
<a class="sourceLine" id="cb2-47" data-line-number="47"><span class="co">#&gt; [1-5] [7-9]</span></a>
<a class="sourceLine" id="cb2-48" data-line-number="48"><span class="co">#&gt; ℹ Smoothing chr16: 10 segments.</span></a>
<a class="sourceLine" id="cb2-49" data-line-number="49"><span class="co">#&gt; [1-6] [8-10]</span></a>
<a class="sourceLine" id="cb2-50" data-line-number="50"><span class="co">#&gt; ℹ Smoothing chr17: 10 segments.</span></a>
<a class="sourceLine" id="cb2-51" data-line-number="51"><span class="co">#&gt; [4-5] [6-7] [8-10]</span></a>
<a class="sourceLine" id="cb2-52" data-line-number="52"><span class="co">#&gt; ℹ Smoothing chr18: 8 segments.</span></a>
<a class="sourceLine" id="cb2-53" data-line-number="53"><span class="co">#&gt; [1-2] [3-8]</span></a>
<a class="sourceLine" id="cb2-54" data-line-number="54"><span class="co">#&gt; ℹ Smoothing chr19: 5 segments.</span></a>
<a class="sourceLine" id="cb2-55" data-line-number="55"><span class="co">#&gt; [2-5]</span></a>
<a class="sourceLine" id="cb2-56" data-line-number="56"><span class="co">#&gt; ℹ Smoothing chr2: 18 segments.</span></a>
<a class="sourceLine" id="cb2-57" data-line-number="57"><span class="co">#&gt; [1-8] [10-12] [14-18]</span></a>
<a class="sourceLine" id="cb2-58" data-line-number="58"><span class="co">#&gt; ℹ Smoothing chr20: 9 segments.</span></a>
<a class="sourceLine" id="cb2-59" data-line-number="59"><span class="co">#&gt; [1-3] [4-9]</span></a>
<a class="sourceLine" id="cb2-60" data-line-number="60"><span class="co">#&gt; ℹ Smoothing chr21: 2 segments.</span></a>
<a class="sourceLine" id="cb2-61" data-line-number="61"><span class="co">#&gt; [1-2]</span></a>
<a class="sourceLine" id="cb2-62" data-line-number="62"><span class="co">#&gt; ℹ Smoothing chr22: 3 segments.</span></a>
<a class="sourceLine" id="cb2-63" data-line-number="63"><span class="co">#&gt; ℹ Smoothing chr3: 19 segments.</span></a>
<a class="sourceLine" id="cb2-64" data-line-number="64"><span class="co">#&gt; [1-7] [8-13] [15-19]</span></a>
<a class="sourceLine" id="cb2-65" data-line-number="65"><span class="co">#&gt; ℹ Smoothing chr4: 8 segments.</span></a>
<a class="sourceLine" id="cb2-66" data-line-number="66"><span class="co">#&gt; [1-4] [5-8]</span></a>
<a class="sourceLine" id="cb2-67" data-line-number="67"><span class="co">#&gt; ℹ Smoothing chr5: 6 segments.</span></a>
<a class="sourceLine" id="cb2-68" data-line-number="68"><span class="co">#&gt; [1-2] [3-5]</span></a>
<a class="sourceLine" id="cb2-69" data-line-number="69"><span class="co">#&gt; ℹ Smoothing chr6: 4 segments.</span></a>
<a class="sourceLine" id="cb2-70" data-line-number="70"><span class="co">#&gt; [1-2] [3-4]</span></a>
<a class="sourceLine" id="cb2-71" data-line-number="71"><span class="co">#&gt; ℹ Smoothing chr7: 46 segments.</span></a>
<a class="sourceLine" id="cb2-72" data-line-number="72"><span class="co">#&gt; [1-5] [9-27] [36-38] [39-41] [43-46]</span></a>
<a class="sourceLine" id="cb2-73" data-line-number="73"><span class="co">#&gt; ℹ Smoothing chr8: 18 segments.</span></a>
<a class="sourceLine" id="cb2-74" data-line-number="74"><span class="co">#&gt; [1-4] [5-11] [12-18]</span></a>
<a class="sourceLine" id="cb2-75" data-line-number="75"><span class="co">#&gt; ℹ Smoothing chr9: 3 segments.</span></a>
<a class="sourceLine" id="cb2-76" data-line-number="76"><span class="co">#&gt; [1-2]</span></a>
<a class="sourceLine" id="cb2-77" data-line-number="77"><span class="co">#&gt; ℹ Smoothing chrX: 6 segments.</span></a>
<a class="sourceLine" id="cb2-78" data-line-number="78"><span class="co">#&gt; [1-2] [3-6]</span></a>
<a class="sourceLine" id="cb2-79" data-line-number="79"><span class="co">#&gt; ✓ Smoothed from 267 to 87 segments with 1e+06 gap ...</span></a>
<a class="sourceLine" id="cb2-80" data-line-number="80"><span class="co">#&gt; ℹ Creating a new CNAqc object. The old object will be retained in the $before_smoothing field.</span></a>
<a class="sourceLine" id="cb2-81" data-line-number="81"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb2-82" data-line-number="82"><span class="co">#&gt; ℹ Using reference genome coordinates for: GRCh38.</span></a>
<a class="sourceLine" id="cb2-83" data-line-number="83"><span class="co">#&gt; ℹ Input n = 12963 mutations for 87 CNA segments (87 clonal, 0 subclonal)</span></a>
<a class="sourceLine" id="cb2-84" data-line-number="84"><span class="co">#&gt; Warning in map_mutations_to_segments(snvs, cna %&gt;% filter(CCF == 1)): [CNAqc] a</span></a>
<a class="sourceLine" id="cb2-85" data-line-number="85"><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></a>
<a class="sourceLine" id="cb2-86" data-line-number="86"><span class="co">#&gt; ✓ Mapped n = 12963 mutations to clonal segments (100% of input)</span></a>
<a class="sourceLine" id="cb2-87" data-line-number="87"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-88" data-line-number="88"><span class="co">#&gt; ── Purity/ ploidy QC via peak detection ─────────────────────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb2-89" data-line-number="89"><span class="co">#&gt; ℹ Requested karyotypes 1:0, 1:1, 2:0, 2:1 and 2:2.</span></a>
<a class="sourceLine" id="cb2-90" data-line-number="90"><span class="co">#&gt; ℹ Found n = 9041 mutations in 2:2, 2:1 (skipping those with n &lt; 648 mutations).</span></a>
<a class="sourceLine" id="cb2-91" data-line-number="91"><span class="co">#&gt; # A tibble: 4 x 12</span></a>
<a class="sourceLine" id="cb2-92" data-line-number="92"><span class="co">#&gt;   mutation_multip… karyotype  peak     x     y counts_per_bin discarded   offset</span></a>
<a class="sourceLine" id="cb2-93" data-line-number="93"><span class="co">#&gt;              &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt; &lt;lgl&gt;        &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb2-94" data-line-number="94"><span class="co">#&gt; 1                2 2:1       0.616  0.61  2.84             50 FALSE      0.00592</span></a>
<a class="sourceLine" id="cb2-95" data-line-number="95"><span class="co">#&gt; 2                1 2:1       0.308  0.31  3.73             65 FALSE     -0.00204</span></a>
<a class="sourceLine" id="cb2-96" data-line-number="96"><span class="co">#&gt; 3                2 2:2       0.471  0.46  8.31            668 FALSE      0.0109 </span></a>
<a class="sourceLine" id="cb2-97" data-line-number="97"><span class="co">#&gt; 4                1 2:2       0.235  0.25  0.92             84 FALSE     -0.0146 </span></a>
<a class="sourceLine" id="cb2-98" data-line-number="98"><span class="co">#&gt; # … with 4 more variables: matched &lt;lgl&gt;, weight &lt;dbl&gt;, score &lt;dbl&gt;, QC &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb2-99" data-line-number="99"><span class="co">#&gt; ✓ Peak detection PASS with r = -0.00117483319779741 and tolerance e = 0.05</span></a>
<a class="sourceLine" id="cb2-100" data-line-number="100"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-101" data-line-number="101"><span class="co">#&gt; ── CCF estimation and QC ────────────────────────────────────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb2-102" data-line-number="102"><span class="co">#&gt; ── Computing mutation multiplicity for karyotype 2:0 using the entropy method. ──────────────────────────</span></a>
<a class="sourceLine" id="cb2-103" data-line-number="103"><span class="co">#&gt; ℹ Expected Binomial peak(s) for these calls (1 and 2 copies): 0.445 and 0.89</span></a>
<a class="sourceLine" id="cb2-104" data-line-number="104"><span class="co">#&gt; ℹ Mixing pre/ post aneuploidy: 0.09 and 0.91</span></a>
<a class="sourceLine" id="cb2-105" data-line-number="105"><span class="co">#&gt; ℹ Not assignamble area: [0.631578947368421; 0.723684210526316]</span></a>
<a class="sourceLine" id="cb2-106" data-line-number="106"><span class="co">#&gt; ── Computing mutation multiplicity for karyotype 2:1 using the entropy method. ──────────────────────────</span></a>
<a class="sourceLine" id="cb2-107" data-line-number="107"><span class="co">#&gt; ℹ Expected Binomial peak(s) for these calls (1 and 2 copies): 0.307958477508651 and 0.615916955017301</span></a>
<a class="sourceLine" id="cb2-108" data-line-number="108"><span class="co">#&gt; ℹ Mixing pre/ post aneuploidy: 0.55 and 0.45</span></a>
<a class="sourceLine" id="cb2-109" data-line-number="109"><span class="co">#&gt; ℹ Not assignamble area: [0.423423423423423; 0.504504504504504]</span></a>
<a class="sourceLine" id="cb2-110" data-line-number="110"><span class="co">#&gt; ── Computing mutation multiplicity for karyotype 2:2 using the entropy method. ──────────────────────────</span></a>
<a class="sourceLine" id="cb2-111" data-line-number="111"><span class="co">#&gt; ℹ Expected Binomial peak(s) for these calls (1 and 2 copies): 0.235449735449735 and 0.470899470899471</span></a>
<a class="sourceLine" id="cb2-112" data-line-number="112"><span class="co">#&gt; ℹ Mixing pre/ post aneuploidy: 0.09 and 0.91</span></a>
<a class="sourceLine" id="cb2-113" data-line-number="113"><span class="co">#&gt; ℹ Not assignamble area: [0.290780141843972; 0.368794326241135]</span></a>
<a class="sourceLine" id="cb2-114" data-line-number="114"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-115" data-line-number="115"><span class="co">#&gt; ── Detecting patterns of overfragmentation ──────────────────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb2-116" data-line-number="116"><span class="co">#&gt; ℹ One-tailed Binomial test: 2 tests, alpha 0.01. Short segments: 0.2% of the reference arm.</span></a>
<a class="sourceLine" id="cb2-117" data-line-number="117"><span class="co">#&gt; ✓ chr7p,  p = 4.52608e-06 ~ 12 segments, 10 short.</span></a>
<a class="sourceLine" id="cb2-118" data-line-number="118"><span class="co">#&gt; ✓ chr12q,  p = 9.21599999999998e-07 ~ 11 segments, 10 short.</span></a>
<a class="sourceLine" id="cb2-119" data-line-number="119"><span class="co">#&gt; ℹ 2 significantly overfragmented chromosome arms (alpha level 0.01).</span></a></code></pre></div>
<div id="extracting-output-information" class="section level2">
<h2 class="hasAnchor">
<a href="#extracting-output-information" class="anchor"></a>Extracting output information</h2>
<p>A special S3 output object is used to store results from this pipeline; the object implements methods for <code>print</code> and <code>plot</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># S3 object</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/class.html">class</a></span>(pipeline)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; [1] "evopipe_qc"</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co"># Summary of the results</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(pipeline)</a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt; ── [ Evoverse ] Example dataset for QC QC pipeline ──────────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt; ── [ CNAqc ]  n = 12963 mutations in 87 segments (87 clonal + 0 subclonal). Genome reference: GRCh38. ───</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt;    2:2  [n = 7478] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { chr19:15292495:15292496:G:A }</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt;    4:2  [n = 1893] ■■■■■■■</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt;    3:2  [n = 1625] ■■■■■■</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt;    2:1  [n = 1563] ■■■■■■  { chr2:179431633:179431634:C:T }</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt;    3:0  [n =  312] ■</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 4.</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; ℹ Mutations annotated have 3 drivers.</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; ✓ QC via peak detection available, score: -0.00117483319779741.</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt; # A tibble: 4 x 12</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt;   mutation_multip… karyotype  peak     x     y counts_per_bin discarded   offset</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">#&gt;              &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt; &lt;lgl&gt;        &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">#&gt; 1                2 2:1       0.616  0.61  2.84             50 FALSE      0.00592</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">#&gt; 2                1 2:1       0.308  0.31  3.73             65 FALSE     -0.00204</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="co">#&gt; 3                2 2:2       0.471  0.46  8.31            668 FALSE      0.0109 </span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="co">#&gt; 4                1 2:2       0.235  0.25  0.92             84 FALSE     -0.0146 </span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="co">#&gt; # … with 4 more variables: matched &lt;lgl&gt;, weight &lt;dbl&gt;, score &lt;dbl&gt;, QC &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="co">#&gt; ✓ Cancer Cell Fraction (CCF) data available for karyotypes: 2:0, 2:1 and 2:2.</span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="co">#&gt; ✓ These segments are smoothed; before smoothing there were 267 segments.</span></a>
<a class="sourceLine" id="cb3-28" data-line-number="28"><span class="co">#&gt; ✓ Arm-level fragmentation analysis: 2 segments overfragmented.</span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="co">#&gt; ──  QC PASS  ────────────────────────────────────────────────────────────────────── PASS rate (%): 100 ──</span></a>
<a class="sourceLine" id="cb3-30" data-line-number="30"><span class="co">#&gt; # A tibble: 5 x 3</span></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"><span class="co">#&gt;   karyotype QC    type </span></a>
<a class="sourceLine" id="cb3-32" data-line-number="32"><span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb3-33" data-line-number="33"><span class="co">#&gt; 1 2:1       PASS  Peaks</span></a>
<a class="sourceLine" id="cb3-34" data-line-number="34"><span class="co">#&gt; 2 2:2       PASS  Peaks</span></a>
<a class="sourceLine" id="cb3-35" data-line-number="35"><span class="co">#&gt; 3 2:0       PASS  CCF  </span></a>
<a class="sourceLine" id="cb3-36" data-line-number="36"><span class="co">#&gt; 4 2:1       PASS  CCF  </span></a>
<a class="sourceLine" id="cb3-37" data-line-number="37"><span class="co">#&gt; 5 2:2       PASS  CCF</span></a></code></pre></div>
<p>The fields <code>pipeline$cnaqc</code> is the <a href="https://caravagn.github.io/CNAqc/">CNAqc</a> object computed by the pipeline; it contains CCF, peaks analysis etc. and can be accessed with the <a href="https://caravagn.github.io/CNAqc/">methods of the package</a>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># CNAqc object</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(pipeline<span class="op">$</span>cnaqc)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt; ── [ CNAqc ]  n = 12963 mutations in 87 segments (87 clonal + 0 subclonal). Genome reference: GRCh38. ───</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt;    2:2  [n = 7478] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { chr19:15292495:15292496:G:A }</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt;    4:2  [n = 1893] ■■■■■■■</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt;    3:2  [n = 1625] ■■■■■■</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt;    2:1  [n = 1563] ■■■■■■  { chr2:179431633:179431634:C:T }</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt;    3:0  [n =  312] ■</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 4.</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt; ℹ Mutations annotated have 3 drivers.</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt; ✓ QC via peak detection available, score: -0.00117483319779741.</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt; # A tibble: 4 x 12</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt;   mutation_multip… karyotype  peak     x     y counts_per_bin discarded   offset</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt;              &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt; &lt;lgl&gt;        &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt; 1                2 2:1       0.616  0.61  2.84             50 FALSE      0.00592</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="co">#&gt; 2                1 2:1       0.308  0.31  3.73             65 FALSE     -0.00204</span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co">#&gt; 3                2 2:2       0.471  0.46  8.31            668 FALSE      0.0109 </span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co">#&gt; 4                1 2:2       0.235  0.25  0.92             84 FALSE     -0.0146 </span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="co">#&gt; # … with 4 more variables: matched &lt;lgl&gt;, weight &lt;dbl&gt;, score &lt;dbl&gt;, QC &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="co">#&gt; ✓ Cancer Cell Fraction (CCF) data available for karyotypes: 2:0, 2:1 and 2:2.</span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="co">#&gt; ✓ These segments are smoothed; before smoothing there were 267 segments.</span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co">#&gt; ✓ Arm-level fragmentation analysis: 2 segments overfragmented.</span></a></code></pre></div>
<p>A summary QC table of the pipeline can be accessed; scores <code>f_PASS</code> and <code>NA_test</code> are percentages of tests that have achieved QC <code>"PASS"</code>, and tests that could not be computed (because of constraints of parameters, or missing data etc.)</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(pipeline<span class="op">$</span>QC)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="co">#&gt; $QC_table</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; # A tibble: 10 x 3</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt;    karyotype QC    type </span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt;  1 1:0       &lt;NA&gt;  Peaks</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt;  2 1:1       &lt;NA&gt;  Peaks</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt;  3 2:0       &lt;NA&gt;  Peaks</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt;  4 2:1       PASS  Peaks</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt;  5 2:2       PASS  Peaks</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt;  6 1:0       &lt;NA&gt;  CCF  </span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt;  7 1:1       &lt;NA&gt;  CCF  </span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt;  8 2:0       PASS  CCF  </span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt;  9 2:1       PASS  CCF  </span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt; 10 2:2       PASS  CCF  </span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="co">#&gt; $f_PASS</span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="co">#&gt; [1] 100</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="co">#&gt; $NA_tests</span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="co">#&gt; [1] 50</span></a></code></pre></div>
<p>And a multi-panel figure is assembled by the S3 <code>plot</code> method, wich reports (by row) CNA segments and genome-wide depth of sequencing, CCF and raw data histograms, results from peak-detection and a strip plot of mutation multiplicity estimates that are used to compute CCFs.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(pipeline)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="co">#&gt; Warning: Removed 6 rows containing missing values (geom_bar).</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt; Warning: Removed 6 rows containing missing values (geom_bar).</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt; Warning: Removed 8 rows containing missing values (geom_bar).</span></a></code></pre></div>
<p><img src="a1_Data_QC_files/figure-html/unnamed-chunk-6-1.png" width="1440"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#pipeline-overview">Pipeline overview</a><ul class="nav nav-pills nav-stacked">
<li><a href="#input-requirements">Input requirements</a></li>
      <li><a href="#pipeline-skeleton">Pipeline skeleton</a></li>
      </ul>
</li>
      <li>
<a href="#example-run">Example run</a><ul class="nav nav-pills nav-stacked">
<li><a href="#extracting-output-information">Extracting output information</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
