<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Timing chromosome aneuploidy • evoverse</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Timing chromosome aneuploidy">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">evoverse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Chr_timing.html">Timing chromosome aneuploidy</a>
    </li>
    <li>
      <a href="../articles/Data_QC.html">Data quality check</a>
    </li>
    <li>
      <a href="../articles/Untitled.html">R Notebook</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/evoverse">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Timing chromosome aneuploidy</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/evoverse/blob/master/vignettes/Chr_timing.Rmd"><code>vignettes/Chr_timing.Rmd</code></a></small>
      <div class="hidden name"><code>Chr_timing.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(evoverse)</a></code></pre></div>
<div id="pipeline" class="section level1">
<h1 class="hasAnchor">
<a href="#pipeline" class="anchor"></a>Pipeline</h1>
<p><img src="pipeline_timing.png" alt="drawing" height="100"></p>
<p><strong>Input requirements.</strong></p>
<ul>
<li>one <a href="https://caravagn.github.io/CNAqc/">CNAqc</a> input dataset,
<ul>
<li>somatic mutations (with read counts data);</li>
<li>absolute copy number segments;</li>
<li>tumour purity.</li>
</ul>
</li>
<li>a list of karyotypes to time, supported ones are <code><a href="https://rdrr.io/r/base/c.html">c('2:0', '2:1', '2:2')</a></code> (default);</li>
<li>pipeline parameters.</li>
</ul>
<p><strong>Pipeline skeletons.</strong></p>
<ol style="list-style-type: decimal">
<li>map mutations to copy number segments, with <a href="https://caravagn.github.io/CNAqc/">CNAqc</a>;</li>
<li>subset mutations according to the input list of karyotypes;</li>
<li>perform <a href="https://caravagn.github.io/mobster/">MOBSTER</a> deconvolution from raw VAF data:
<ul>
<li>attempt a first run using the ellipsis parameters;</li>
<li>use the timing classifier of <code>evoverse</code> to determine QC PASS/ FAIL status per karyotype;</li>
<li>attempt a second run with karyotyepes that fail QC, using <code>mobster</code>’s default parameters;</li>
<li>re-assess QC for the re-run cases.</li>
</ul>
</li>
</ol>
<ul>
<li>create output summary and assignments tables.</li>
</ul>
</div>
<div id="example-run" class="section level1">
<h1 class="hasAnchor">
<a href="#example-run" class="anchor"></a>Example run</h1>
<p>Example input data from the example CNAqc dataset</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Use example data in the CNAqc package</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(example_dataset_CNAqc, <span class="dt">package =</span> <span class="st">'CNAqc'</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"></a>
<a class="sourceLine" id="cb2-4" data-line-number="4">pipeline =<span class="st"> </span><span class="kw"><a href="../reference/pipeline_chromosome_timing.html">pipeline_chromosome_timing</a></span>(</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">  <span class="dt">mutations =</span> CNAqc<span class="op">::</span>example_dataset_CNAqc<span class="op">$</span>snvs,  <span class="co"># Somatic mutations</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6">  <span class="dt">cna =</span> CNAqc<span class="op">::</span>example_dataset_CNAqc<span class="op">$</span>cna,         <span class="co"># Copy number segments</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">  <span class="dt">purity =</span> CNAqc<span class="op">::</span>example_dataset_CNAqc<span class="op">$</span>purity,   <span class="co"># Sample purity</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8">  <span class="dt">reference =</span> <span class="st">'GRCh38'</span>,                           <span class="co"># Genome reference</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9">  <span class="dt">N_max =</span> <span class="dv">10000</span>,                                  <span class="co"># Downsample karyotypes with &gt; 10K mutations (none here)</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10">  <span class="dt">min_muts =</span> <span class="dv">150</span>,                                 <span class="co"># Minimum number of mutations per karyotype</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11">  <span class="dt">min_VAF =</span> <span class="fl">0.05</span>,                                 <span class="co"># Minimum VAF </span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12">  <span class="dt">description =</span> <span class="st">"Example timing dataset"</span>,         <span class="co"># Dataset description</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13">  <span class="dt">auto_setup =</span> <span class="st">'FAST'</span>                             <span class="co"># Ellipsis (parameters forwarded to each mobster first run)</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14">)</a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="co">#&gt;  [ Evoverse Chromosomal timing pipeline ]</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="co">#&gt; ℹ Loading input data</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19">                    </a>
<a class="sourceLine" id="cb2-20" data-line-number="20">── Preparing input <span class="cf">for</span> deconvolution ─────────────────────────────────────────────────────────────────</a>
<a class="sourceLine" id="cb2-21" data-line-number="21"><span class="co">#&gt; ℹ Loading input data</span></a>
<a class="sourceLine" id="cb2-22" data-line-number="22"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-23" data-line-number="23">                    </a>
<a class="sourceLine" id="cb2-24" data-line-number="24">ℹ Removing <span class="dv">80</span> mutations with VAF <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>.<span class="fl">05.</span></a>
<a class="sourceLine" id="cb2-25" data-line-number="25"><span class="co">#&gt; ℹ Loading input data</span></a>
<a class="sourceLine" id="cb2-26" data-line-number="26"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb2-27" data-line-number="27"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-28" data-line-number="28">                    </a>
<a class="sourceLine" id="cb2-29" data-line-number="29">ℹ Using reference genome coordinates <span class="cf">for</span><span class="op">:</span><span class="st"> </span>GRCh38.</a>
<a class="sourceLine" id="cb2-30" data-line-number="30"><span class="co">#&gt; ℹ Loading input data</span></a>
<a class="sourceLine" id="cb2-31" data-line-number="31">                    </a>
<a class="sourceLine" id="cb2-32" data-line-number="32">ℹ Drivers are annotated, but <span class="st">'gene'</span> column is missing, using mutation location.</a>
<a class="sourceLine" id="cb2-33" data-line-number="33"><span class="co">#&gt; ℹ Loading input data</span></a>
<a class="sourceLine" id="cb2-34" data-line-number="34">                    </a>
<a class="sourceLine" id="cb2-35" data-line-number="35"><span class="op">!</span><span class="st"> </span>Missing CCF column from CNA calls, adding CCF =<span class="st"> </span><span class="dv">1</span> assuming clonal CNA calls.</a>
<a class="sourceLine" id="cb2-36" data-line-number="36"><span class="co">#&gt; ℹ Loading input data</span></a>
<a class="sourceLine" id="cb2-37" data-line-number="37">                    </a>
<a class="sourceLine" id="cb2-38" data-line-number="38">ℹ Input n =<span class="st"> </span><span class="dv">13053</span> mutations <span class="cf">for</span> <span class="dv">267</span> CNA <span class="kw"><a href="https://rdrr.io/r/graphics/segments.html">segments</a></span> (<span class="dv">267</span> clonal, <span class="dv">0</span> subclonal)</a>
<a class="sourceLine" id="cb2-39" data-line-number="39"><span class="co">#&gt; ℹ Loading input data</span></a>
<a class="sourceLine" id="cb2-40" data-line-number="40">                    </a>
<a class="sourceLine" id="cb2-41" data-line-number="41">x n =<span class="st"> </span><span class="dv">172</span> mutations  cannot be mapped to segments and will be removed.</a>
<a class="sourceLine" id="cb2-42" data-line-number="42"><span class="co">#&gt; ℹ Loading input data</span></a>
<a class="sourceLine" id="cb2-43" data-line-number="43"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-44" data-line-number="44">                    </a>
<a class="sourceLine" id="cb2-45" data-line-number="45">✓ Mapped n =<span class="st"> </span><span class="dv">12881</span> mutations to clonal <span class="kw"><a href="https://rdrr.io/r/graphics/segments.html">segments</a></span> (<span class="dv">100</span>% of input)</a>
<a class="sourceLine" id="cb2-46" data-line-number="46"><span class="co">#&gt; ℹ Loading input data</span></a>
<a class="sourceLine" id="cb2-47" data-line-number="47">                    </a>
<a class="sourceLine" id="cb2-48" data-line-number="48">── [ CNAqc ]  n =<span class="st"> </span><span class="dv">12881</span> mutations <span class="cf">in</span> <span class="dv">267</span> <span class="kw"><a href="https://rdrr.io/r/graphics/segments.html">segments</a></span> (<span class="dv">267</span> clonal <span class="op">+</span><span class="st"> </span><span class="dv">0</span> subclonal). Genome reference<span class="op">:</span><span class="st"> </span>GRCh38</a>
<a class="sourceLine" id="cb2-49" data-line-number="49"><span class="co">#&gt; ℹ Loading input data</span></a>
<a class="sourceLine" id="cb2-50" data-line-number="50"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-51" data-line-number="51"><span class="co">#&gt;    2:2  [n = 7437] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { chr19:15292495:15292496:G:A }</span></a>
<a class="sourceLine" id="cb2-52" data-line-number="52"><span class="co">#&gt;    4:2  [n = 1873] ■■■■■■■</span></a>
<a class="sourceLine" id="cb2-53" data-line-number="53"><span class="co">#&gt;    3:2  [n = 1612] ■■■■■■</span></a>
<a class="sourceLine" id="cb2-54" data-line-number="54"><span class="co">#&gt;    2:1  [n = 1557] ■■■■■■  { chr2:179431633:179431634:C:T }</span></a>
<a class="sourceLine" id="cb2-55" data-line-number="55"><span class="co">#&gt;    3:0  [n =  310] ■</span></a>
<a class="sourceLine" id="cb2-56" data-line-number="56"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-57" data-line-number="57">                    </a>
<a class="sourceLine" id="cb2-58" data-line-number="58">ℹ Sample Purity<span class="op">:</span><span class="st"> </span><span class="dv">89</span>% <span class="op">~</span><span class="st"> </span>Ploidy<span class="op">:</span><span class="st"> </span><span class="fl">4.</span></a>
<a class="sourceLine" id="cb2-59" data-line-number="59"><span class="co">#&gt; ℹ Loading input data</span></a>
<a class="sourceLine" id="cb2-60" data-line-number="60"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-61" data-line-number="61">                    </a>
<a class="sourceLine" id="cb2-62" data-line-number="62">ℹ Mutations annotated have <span class="dv">3</span> drivers.</a>
<a class="sourceLine" id="cb2-63" data-line-number="63"><span class="co">#&gt; ℹ Loading input data</span></a>
<a class="sourceLine" id="cb2-64" data-line-number="64">                    </a>
<a class="sourceLine" id="cb2-65" data-line-number="65">✓ Loading input data ... done</a>
<a class="sourceLine" id="cb2-66" data-line-number="66"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-67" data-line-number="67"><span class="co">#&gt; ── MOBSTER clustering mutations with karyotype 2:0 ───────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb2-68" data-line-number="68"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-69" data-line-number="69"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb2-70" data-line-number="70"><span class="co">#&gt; ℹ Using reference genome coordinates for: GRCh38.</span></a>
<a class="sourceLine" id="cb2-71" data-line-number="71"><span class="co">#&gt; ℹ Input n = 12881 mutations for 9 CNA segments (9 clonal, 0 subclonal)</span></a>
<a class="sourceLine" id="cb2-72" data-line-number="72"><span class="co">#&gt; Warning in map_mutations_to_segments(snvs, cna %&gt;% filter(CCF == 1)): [CNAqc] a</span></a>
<a class="sourceLine" id="cb2-73" data-line-number="73"><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></a>
<a class="sourceLine" id="cb2-74" data-line-number="74"><span class="co">#&gt; x n = 12800 mutations  cannot be mapped to segments and will be removed.</span></a>
<a class="sourceLine" id="cb2-75" data-line-number="75"><span class="co">#&gt; ✓ Mapped n = 81 mutations to clonal segments (100% of input)</span></a>
<a class="sourceLine" id="cb2-76" data-line-number="76"><span class="co">#&gt; ! Less than 150 mutations, will no fit.</span></a>
<a class="sourceLine" id="cb2-77" data-line-number="77"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-78" data-line-number="78"><span class="co">#&gt; ── MOBSTER clustering mutations with karyotype 2:1 ───────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb2-79" data-line-number="79"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-80" data-line-number="80"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb2-81" data-line-number="81"><span class="co">#&gt; ℹ Using reference genome coordinates for: GRCh38.</span></a>
<a class="sourceLine" id="cb2-82" data-line-number="82"><span class="co">#&gt; ℹ Input n = 12881 mutations for 49 CNA segments (49 clonal, 0 subclonal)</span></a>
<a class="sourceLine" id="cb2-83" data-line-number="83"><span class="co">#&gt; Warning in map_mutations_to_segments(snvs, cna %&gt;% filter(CCF == 1)): [CNAqc] a</span></a>
<a class="sourceLine" id="cb2-84" data-line-number="84"><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></a>
<a class="sourceLine" id="cb2-85" data-line-number="85"><span class="co">#&gt; x n = 11324 mutations  cannot be mapped to segments and will be removed.</span></a>
<a class="sourceLine" id="cb2-86" data-line-number="86"><span class="co">#&gt; ✓ Mapped n = 1557 mutations to clonal segments (100% of input)</span></a>
<a class="sourceLine" id="cb2-87" data-line-number="87"><span class="co">#&gt;  [ MOBSTER fit ]</span></a>
<a class="sourceLine" id="cb2-88" data-line-number="88"><span class="co">#&gt; ✔ Loaded input data, n = 1557.</span></a>
<a class="sourceLine" id="cb2-89" data-line-number="89"><span class="co">#&gt; ❯ n = 1557. Mixture with k = 1,2 Beta(s). Pareto tail: TRUE and FALSE. Output clusters with π &gt; 0.02</span></a>
<a class="sourceLine" id="cb2-90" data-line-number="90"><span class="co">#&gt; and n &gt; 10.</span></a>
<a class="sourceLine" id="cb2-91" data-line-number="91"><span class="co">#&gt; ! mobster automatic setup FAST for the analysis.</span></a>
<a class="sourceLine" id="cb2-92" data-line-number="92"><span class="co">#&gt; ❯ Scoring (without parallel) 2 x 2 x 2 = 8 models by reICL.</span></a>
<a class="sourceLine" id="cb2-93" data-line-number="93"><span class="co">#&gt; ℹ MOBSTER fits completed in 6.1s.</span></a>
<a class="sourceLine" id="cb2-94" data-line-number="94"><span class="co">#&gt; ── [ MOBSTER ] Raw VAF for 2:1 n = 1557 with k = 1 Beta(s) without tail ──────────────────────────────</span></a>
<a class="sourceLine" id="cb2-95" data-line-number="95"><span class="co">#&gt; ● Clusters: π = 100% [C1], with π &gt; 0.</span></a>
<a class="sourceLine" id="cb2-96" data-line-number="96"><span class="co">#&gt; ✖ No tail fit.</span></a>
<a class="sourceLine" id="cb2-97" data-line-number="97"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-98" data-line-number="98"><span class="co">#&gt; ● Beta C1 [n = 1557, 100%] with mean = 0.42.</span></a>
<a class="sourceLine" id="cb2-99" data-line-number="99"><span class="co">#&gt; ℹ Score(s): NLL = -660.35; ICL = -1298.65 (-1298.65), H = 0 (0). Fit interrupted by MM in 100 steps.</span></a>
<a class="sourceLine" id="cb2-100" data-line-number="100"><span class="co">#&gt; Warning: Unknown or uninitialised column: 'Scale'.</span></a>
<a class="sourceLine" id="cb2-101" data-line-number="101"><span class="co">#&gt; Warning: Unknown or uninitialised column: 'Shape'.</span></a>
<a class="sourceLine" id="cb2-102" data-line-number="102"><span class="co">#&gt; Warning: Unknown or uninitialised column: 'Scale'.</span></a>
<a class="sourceLine" id="cb2-103" data-line-number="103"><span class="co">#&gt; Warning: Unknown or uninitialised column: 'Shape'.</span></a>
<a class="sourceLine" id="cb2-104" data-line-number="104"><span class="co">#&gt; x Raw VAF for 2:1: QC FAIL. p = 0</span></a>
<a class="sourceLine" id="cb2-105" data-line-number="105"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-106" data-line-number="106"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-107" data-line-number="107"><span class="co">#&gt;         ┌──────────────────────────────────────────────────────────────────────────────────────┐</span></a>
<a class="sourceLine" id="cb2-108" data-line-number="108"><span class="co">#&gt;         │                                                                                      │</span></a>
<a class="sourceLine" id="cb2-109" data-line-number="109"><span class="co">#&gt;         │   Auto-QC (type 'T') failed, attempting another single run with default parameters   │</span></a>
<a class="sourceLine" id="cb2-110" data-line-number="110"><span class="co">#&gt;         │                                                                                      │</span></a>
<a class="sourceLine" id="cb2-111" data-line-number="111"><span class="co">#&gt;         └──────────────────────────────────────────────────────────────────────────────────────┘</span></a>
<a class="sourceLine" id="cb2-112" data-line-number="112"><span class="co">#&gt;  [ MOBSTER fit ]</span></a>
<a class="sourceLine" id="cb2-113" data-line-number="113"><span class="co">#&gt; ✔ Loaded input data, n = 1557.</span></a>
<a class="sourceLine" id="cb2-114" data-line-number="114"><span class="co">#&gt; ❯ n = 1557. Mixture with k = 1,2,3 Beta(s). Pareto tail: TRUE and FALSE. Output clusters with π &gt;</span></a>
<a class="sourceLine" id="cb2-115" data-line-number="115"><span class="co">#&gt; 0.02 and n &gt; 10.</span></a>
<a class="sourceLine" id="cb2-116" data-line-number="116"><span class="co">#&gt; ❯ Custom fit by Moments-matching in up to 250 steps, with ε = 1e-10 and peaks initialisation.</span></a>
<a class="sourceLine" id="cb2-117" data-line-number="117"><span class="co">#&gt; ❯ Scoring (without parallel) 5 x 3 x 2 = 30 models by reICL.</span></a>
<a class="sourceLine" id="cb2-118" data-line-number="118"><span class="co">#&gt; ℹ MOBSTER fits completed in 51.7s.</span></a>
<a class="sourceLine" id="cb2-119" data-line-number="119"><span class="co">#&gt; ── [ MOBSTER ] Raw VAF for 2:1 n = 1557 with k = 2 Beta(s) and a tail ────────────────────────────────</span></a>
<a class="sourceLine" id="cb2-120" data-line-number="120"><span class="co">#&gt; ● Clusters: π = 53% [C2], 43% [C1] and 4% [Tail], with π &gt; 0.</span></a>
<a class="sourceLine" id="cb2-121" data-line-number="121"><span class="co">#&gt; ● Tail [n = 50, 3.8%] with alpha = 1.2.</span></a>
<a class="sourceLine" id="cb2-122" data-line-number="122"><span class="co">#&gt; ● Beta C1 [n = 677, 43%] with mean = 0.6.</span></a>
<a class="sourceLine" id="cb2-123" data-line-number="123"><span class="co">#&gt; ● Beta C2 [n = 830, 53%] with mean = 0.31.</span></a>
<a class="sourceLine" id="cb2-124" data-line-number="124"><span class="co">#&gt; ℹ Score(s): NLL = -1284.16; ICL = -2436.76 (-2496.86), H = 65.41 (5.31). Fit converged by MM in 42</span></a>
<a class="sourceLine" id="cb2-125" data-line-number="125"><span class="co">#&gt; steps.</span></a>
<a class="sourceLine" id="cb2-126" data-line-number="126"><span class="co">#&gt; ✓ Raw VAF for 2:1: QC PASS. p = 0.993</span></a>
<a class="sourceLine" id="cb2-127" data-line-number="127"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-128" data-line-number="128"><span class="co">#&gt; ── MOBSTER clustering mutations with karyotype 2:2 ───────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb2-129" data-line-number="129"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-130" data-line-number="130"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb2-131" data-line-number="131"><span class="co">#&gt; ℹ Using reference genome coordinates for: GRCh38.</span></a>
<a class="sourceLine" id="cb2-132" data-line-number="132"><span class="co">#&gt; ℹ Input n = 12881 mutations for 80 CNA segments (80 clonal, 0 subclonal)</span></a>
<a class="sourceLine" id="cb2-133" data-line-number="133"><span class="co">#&gt; Warning in map_mutations_to_segments(snvs, cna %&gt;% filter(CCF == 1)): [CNAqc] a</span></a>
<a class="sourceLine" id="cb2-134" data-line-number="134"><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></a>
<a class="sourceLine" id="cb2-135" data-line-number="135"><span class="co">#&gt; x n = 5444 mutations  cannot be mapped to segments and will be removed.</span></a>
<a class="sourceLine" id="cb2-136" data-line-number="136"><span class="co">#&gt; ✓ Mapped n = 7437 mutations to clonal segments (100% of input)</span></a>
<a class="sourceLine" id="cb2-137" data-line-number="137"><span class="co">#&gt;  [ MOBSTER fit ]</span></a>
<a class="sourceLine" id="cb2-138" data-line-number="138"><span class="co">#&gt; ✔ Loaded input data, n = 7434.</span></a>
<a class="sourceLine" id="cb2-139" data-line-number="139"><span class="co">#&gt; ❯ n = 7434. Mixture with k = 1,2 Beta(s). Pareto tail: TRUE and FALSE. Output clusters with π &gt; 0.02</span></a>
<a class="sourceLine" id="cb2-140" data-line-number="140"><span class="co">#&gt; and n &gt; 10.</span></a>
<a class="sourceLine" id="cb2-141" data-line-number="141"><span class="co">#&gt; ! mobster automatic setup FAST for the analysis.</span></a>
<a class="sourceLine" id="cb2-142" data-line-number="142"><span class="co">#&gt; ❯ Scoring (without parallel) 2 x 2 x 2 = 8 models by reICL.</span></a>
<a class="sourceLine" id="cb2-143" data-line-number="143"><span class="co">#&gt; ℹ MOBSTER fits completed in 8.8s.</span></a>
<a class="sourceLine" id="cb2-144" data-line-number="144"><span class="co">#&gt; ── [ MOBSTER ] Raw VAF for 2:2 n = 7434 with k = 2 Beta(s) and a tail ────────────────────────────────</span></a>
<a class="sourceLine" id="cb2-145" data-line-number="145"><span class="co">#&gt; ● Clusters: π = 86% [C1], 8% [C2] and 5% [Tail], with π &gt; 0.</span></a>
<a class="sourceLine" id="cb2-146" data-line-number="146"><span class="co">#&gt; ● Tail [n = 335, 5.5%] with alpha = 1.3.</span></a>
<a class="sourceLine" id="cb2-147" data-line-number="147"><span class="co">#&gt; ● Beta C1 [n = 6440, 86%] with mean = 0.46.</span></a>
<a class="sourceLine" id="cb2-148" data-line-number="148"><span class="co">#&gt; ● Beta C2 [n = 659, 8%] with mean = 0.23.</span></a>
<a class="sourceLine" id="cb2-149" data-line-number="149"><span class="co">#&gt; ℹ Score(s): NLL = -9480.88; ICL = -18514.11 (-18847.52), H = 367.42 (34.02). Fit converged by MM in</span></a>
<a class="sourceLine" id="cb2-150" data-line-number="150"><span class="co">#&gt; 56 steps.</span></a>
<a class="sourceLine" id="cb2-151" data-line-number="151"><span class="co">#&gt; ✓ Raw VAF for 2:2: QC PASS. p = 0.905</span></a>
<a class="sourceLine" id="cb2-152" data-line-number="152"><span class="co">#&gt; ℹ Pipeline results assembly</span></a>
<a class="sourceLine" id="cb2-153" data-line-number="153"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-154" data-line-number="154">                           </a>
<a class="sourceLine" id="cb2-155" data-line-number="155">✓ Pipeline results assembly ... done</a></code></pre></div>
<p>S3 output object</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(pipeline)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co">#&gt; ── [ Evoverse ] Example timing dataset Chromosome timing pipeline ────────────────────────────────────</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; ── [ CNAqc ]  n = 12881 mutations in 267 segments (267 clonal + 0 subclonal). Genome reference: GRCh38</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt;    2:2  [n = 7437] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { chr19:15292495:15292496:G:A }</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt;    4:2  [n = 1873] ■■■■■■■</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt;    3:2  [n = 1612] ■■■■■■</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt;    2:1  [n = 1557] ■■■■■■  { chr2:179431633:179431634:C:T }</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt;    3:0  [n =  310] ■</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 4.</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt; ℹ Mutations annotated have 3 drivers.</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt; ──  QC PASS  ────────────────────────────────────────────────────────────────────── PASS rate (%):  ──</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt; # A tibble: 2 x 4</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt;   karyotype QC    QC_prob QC_type                </span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;                  </span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; 1 2:1       PASS    0.993 timing_logistic_GEL_CRC</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; 2 2:2       PASS    0.905 timing_logistic_GEL_CRC</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt;   LOG  2020-04-25 19:29:18. evoverse pipeline for chromosome timing: with fits TRUE</span></a></code></pre></div>
<p>Output fields</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># MOBSTER fits per karyotype</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(pipeline<span class="op">$</span>mobster)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt; $`2:1`</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt; ── [ MOBSTER ] Raw VAF for 2:1 n = 1557 with k = 2 Beta(s) and a tail ────────────────────────────────</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt; ● Clusters: π = 53% [C2], 43% [C1] and 4% [Tail], with π &gt; 0.</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt; ● Tail [n = 50, 3.8%] with alpha = 1.2.</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt; ● Beta C1 [n = 677, 43%] with mean = 0.6.</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; ● Beta C2 [n = 830, 53%] with mean = 0.31.</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt; ℹ Score(s): NLL = -1284.16; ICL = -2436.76 (-2496.86), H = 65.41 (5.31). Fit converged by MM in 42</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt; steps.</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt; $`2:2`</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt; ── [ MOBSTER ] Raw VAF for 2:2 n = 7434 with k = 2 Beta(s) and a tail ────────────────────────────────</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt; ● Clusters: π = 86% [C1], 8% [C2] and 5% [Tail], with π &gt; 0.</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt; ● Tail [n = 335, 5.5%] with alpha = 1.3.</span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt; ● Beta C1 [n = 6440, 86%] with mean = 0.46.</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="co">#&gt; ● Beta C2 [n = 659, 8%] with mean = 0.23.</span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co">#&gt; ℹ Score(s): NLL = -9480.88; ICL = -18514.11 (-18847.52), H = 367.42 (34.02). Fit converged by MM in</span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co">#&gt; 56 steps.</span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="co"># Clustering assignments</span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(pipeline<span class="op">$</span>table<span class="op">$</span>clustering_assignments)</a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co">#&gt; # A tibble: 8,991 x 20</span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="co">#&gt;    chr     from     to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION</span></a>
<a class="sourceLine" id="cb4-25" data-line-number="25"><span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb4-26" data-line-number="26"><span class="co">#&gt;  1 chr2  3.58e5 3.58e5 C     A     PASS     104    58 0.558  intergenic      </span></a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="co">#&gt;  2 chr2  9.09e5 9.09e5 A     G     PASS      15     7 0.467  ncRNA_intronic  </span></a>
<a class="sourceLine" id="cb4-28" data-line-number="28"><span class="co">#&gt;  3 chr2  1.04e6 1.04e6 C     T     PASS      93    57 0.613  intronic        </span></a>
<a class="sourceLine" id="cb4-29" data-line-number="29"><span class="co">#&gt;  4 chr2  1.33e6 1.33e6 A     T     PASS     104    55 0.529  intronic        </span></a>
<a class="sourceLine" id="cb4-30" data-line-number="30"><span class="co">#&gt;  5 chr2  1.52e6 1.52e6 C     T     PASS      90    50 0.556  intronic        </span></a>
<a class="sourceLine" id="cb4-31" data-line-number="31"><span class="co">#&gt;  6 chr2  2.20e6 2.20e6 G     T     PASS     125    42 0.336  intronic        </span></a>
<a class="sourceLine" id="cb4-32" data-line-number="32"><span class="co">#&gt;  7 chr2  2.90e6 2.90e6 C     T     PASS     109    58 0.532  downstream      </span></a>
<a class="sourceLine" id="cb4-33" data-line-number="33"><span class="co">#&gt;  8 chr2  3.13e6 3.13e6 A     G     PASS     134    35 0.261  ncRNA_intronic  </span></a>
<a class="sourceLine" id="cb4-34" data-line-number="34"><span class="co">#&gt;  9 chr2  3.83e6 3.83e6 A     T     PASS     120    68 0.567  intergenic      </span></a>
<a class="sourceLine" id="cb4-35" data-line-number="35"><span class="co">#&gt; 10 chr2  3.88e6 3.88e6 T     A     PASS     126    10 0.0794 intergenic      </span></a>
<a class="sourceLine" id="cb4-36" data-line-number="36"><span class="co">#&gt; # … with 8,981 more rows, and 10 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;,</span></a>
<a class="sourceLine" id="cb4-37" data-line-number="37"><span class="co">#&gt; #   driver_label &lt;chr&gt;, gene &lt;chr&gt;, karyotype &lt;chr&gt;, segment_id &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb4-38" data-line-number="38"><span class="co">#&gt; #   mobster_cluster &lt;chr&gt;, Tail &lt;dbl&gt;, C1 &lt;dbl&gt;, C2 &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-39" data-line-number="39"></a>
<a class="sourceLine" id="cb4-40" data-line-number="40"><span class="co"># Summary QC</span></a>
<a class="sourceLine" id="cb4-41" data-line-number="41"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(pipeline<span class="op">$</span>table<span class="op">$</span>summary)</a>
<a class="sourceLine" id="cb4-42" data-line-number="42"><span class="co">#&gt; # A tibble: 2 x 43</span></a>
<a class="sourceLine" id="cb4-43" data-line-number="43"><span class="co">#&gt;   K_beta Mean_C1 Mean_C2 Mean_Tail     N  N_C1  N_C2 N_Tail pi_C1  pi_C2 pi_Tail</span></a>
<a class="sourceLine" id="cb4-44" data-line-number="44"><span class="co">#&gt;    &lt;int&gt;   &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-45" data-line-number="45"><span class="co">#&gt; 1      2   0.601   0.305     0.300  1557   677   830     50 0.433 0.529   0.0385</span></a>
<a class="sourceLine" id="cb4-46" data-line-number="46"><span class="co">#&gt; 2      2   0.462   0.228     0.226  7434  6440   659    335 0.863 0.0827  0.0546</span></a>
<a class="sourceLine" id="cb4-47" data-line-number="47"><span class="co">#&gt; # … with 32 more variables: rcc_C1 &lt;lgl&gt;, rcc_C2 &lt;lgl&gt;, rcc_Tail &lt;lgl&gt;,</span></a>
<a class="sourceLine" id="cb4-48" data-line-number="48"><span class="co">#&gt; #   Scale_Tail &lt;dbl&gt;, Shape_Tail &lt;dbl&gt;, tail &lt;lgl&gt;, Variance_C1 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb4-49" data-line-number="49"><span class="co">#&gt; #   Variance_C2 &lt;dbl&gt;, Variance_Tail &lt;dbl&gt;, NLL &lt;dbl&gt;, BIC &lt;dbl&gt;, AIC &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb4-50" data-line-number="50"><span class="co">#&gt; #   entropy &lt;dbl&gt;, ICL &lt;dbl&gt;, reduced.entropy &lt;dbl&gt;, reICL &lt;dbl&gt;, size &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb4-51" data-line-number="51"><span class="co">#&gt; #   sse_total &lt;dbl&gt;, sse_0_0.1 &lt;dbl&gt;, sse_0.1_0.2 &lt;dbl&gt;, sse_0.2_0.3 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb4-52" data-line-number="52"><span class="co">#&gt; #   sse_0.3_0.4 &lt;dbl&gt;, sse_0.4_0.5 &lt;dbl&gt;, sse_0.5_0.6 &lt;dbl&gt;, sse_0.6_0.7 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb4-53" data-line-number="53"><span class="co">#&gt; #   sse_0.7_0.8 &lt;dbl&gt;, sse_0.8_0.9 &lt;dbl&gt;, sse_0.9_1 &lt;dbl&gt;, QC &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb4-54" data-line-number="54"><span class="co">#&gt; #   QC_prob &lt;dbl&gt;, QC_type &lt;chr&gt;, karyotype &lt;chr&gt;</span></a></code></pre></div>
<p>One-page plotting function (assembles a multi-panel figure)</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(pipeline)</a></code></pre></div>
<p><img src="Chr_timing_files/figure-html/unnamed-chunk-5-1.png" width="1440"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#pipeline">Pipeline</a></li>
      <li><a href="#example-run">Example run</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
