<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Subclonal deconvolution of MSeq multi-region WGS colorectal cancer data • evoverse</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Subclonal deconvolution of MSeq multi-region WGS colorectal cancer data">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">evoverse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/evoverse.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Colorectal_Cross.html">Subclonal deconvolution of MSeq multi-region WGS colorectal cancer data</a>
    </li>
    <li>
      <a href="../articles/packages.html">Packages in the `evoverse`</a>
    </li>
    <li>
      <a href="../articles/pipelines.html">Pipeline available with the `evoverse` packages</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/evoverse">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Subclonal deconvolution of MSeq multi-region WGS colorectal cancer data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/evoverse/blob/master/vignettes/Colorectal_Cross.Rmd"><code>vignettes/Colorectal_Cross.Rmd</code></a></small>
      <div class="hidden name"><code>Colorectal_Cross.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(evoverse)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="co">#&gt;  [ CNAqc - Copy Number Alteration quality check ] </span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">#&gt; GitHub :  caravagn/CNAqc </span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co">#&gt;    WWW :  https://caravagn.github.io/CNAqc/ </span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co">#&gt;  &gt; CNAqc is part of the "evoverse" [https://bit.ly/2orn94e] - a collection of packages to implement Cancer Evolution analyses from cancer sequencing data.</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="co">#&gt;  [ MOBSTER - Model-based clustering in cancer ] </span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="co">#&gt; GitHub :  caravagn/mobster </span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="co">#&gt;    WWW :  https://caravagn.github.io/mobster/ </span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="co">#&gt;  &gt; MOBSTER is part of the "evoverse" [https://bit.ly/2orn94e] - a collection of packages to implement Cancer Evolution analyses from cancer sequencing data.</span></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(evoverse.datasets)</a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="co">#&gt; Available datasets ~ use data('xxx', package='REVOLVER_datasets') to load dataset 'xxx'</span></a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="co">#&gt; ◉ TRACERx_NEJM_2017 Mutations from TRACERx lung (PMID: 28445112). n = 99 patients, multi-region WES, CCF. </span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="co">#&gt; ◉ TRACERx_NEJM_2017_REVOLVER REVOLVER analysis of TRACERx_NEJM_2017. </span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="co">#&gt; ◉ YATES_BREAST_NATMED_2015 Mutations from breast cancers (PMID: 26099045). n = 50 patients, multi-region WES, binary. </span></a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="co">#&gt; ◉ CROSS_CRC_ADENOCARCINOMA_NATECOEVO_2018 Mutations from colorectal adenocarcinomas (PMID: 30177804). n = 19 patients, multi-region WES, binary. </span></a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="co">#&gt; ◉ MSEQ_CRC_ADENOCARCINOMA_SET6 Mutations, copy number and purity from the Set6 multi-region colorectal adenocarcinoma sample (doi:10.1101/586560). n = 1 patient, multi-region WGS </span></a>
<a class="sourceLine" id="cb1-23" data-line-number="23"><span class="co">#&gt; ◉ MSEQ_CRC_ADENOCARCINOMA_SET7 Mutations, copy number and purity from the Set7 multi-region colorectal adenocarcinoma sample (doi:10.1101/586560). n = 1 patient, multi-region WGS</span></a></code></pre></div>
<p>This vignette describes the analysis of the MSeq data released with the paper</p>
<ul>
<li>Caravagna et al. Model-based tumor subclonal reconstruction. biorxiv 2019, <a href="https://www.biorxiv.org/content/10.1101/586560v1">doi:10.1101/586560</a>.</li>
</ul>
<p>The data is also released as Excel Supplementary Table with the paper.</p>
<div id="available-data" class="section level3">
<h3 class="hasAnchor">
<a href="#available-data" class="anchor"></a>Available data</h3>
<p>Mutation, copy nunmber ad purity estimates for both analysed multi-region samples - <code>Set6</code> and <code>Set7</code> - are part of the data released with the <code>evoverse.datasets</code> <a href="https://github.com/caravagn/evoverse.datasets">package</a>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># All available data</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">evoverse.datasets<span class="op">::</span><span class="kw">available_data</span>()</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">#&gt; Available datasets ~ use data('xxx', package='REVOLVER_datasets') to load dataset 'xxx'</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co">#&gt; ◉ TRACERx_NEJM_2017 Mutations from TRACERx lung (PMID: 28445112). n = 99 patients, multi-region WES, CCF. </span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co">#&gt; ◉ TRACERx_NEJM_2017_REVOLVER REVOLVER analysis of TRACERx_NEJM_2017. </span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">#&gt; ◉ YATES_BREAST_NATMED_2015 Mutations from breast cancers (PMID: 26099045). n = 50 patients, multi-region WES, binary. </span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co">#&gt; ◉ CROSS_CRC_ADENOCARCINOMA_NATECOEVO_2018 Mutations from colorectal adenocarcinomas (PMID: 30177804). n = 19 patients, multi-region WES, binary. </span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co">#&gt; ◉ MSEQ_CRC_ADENOCARCINOMA_SET6 Mutations, copy number and purity from the Set6 multi-region colorectal adenocarcinoma sample (doi:10.1101/586560). n = 1 patient, multi-region WGS </span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co">#&gt; ◉ MSEQ_CRC_ADENOCARCINOMA_SET7 Mutations, copy number and purity from the Set7 multi-region colorectal adenocarcinoma sample (doi:10.1101/586560). n = 1 patient, multi-region WGS</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="co"># Set7</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">'MSEQ_CRC_ADENOCARCINOMA_SET7'</span>, <span class="dt">package =</span> <span class="st">'evoverse.datasets'</span>)</a></code></pre></div>
<p>Sneak peek to mutations, copy number, purity data and sample estimates</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(MSEQ_CRC_ADENOCARCINOMA_SET7)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co">#&gt; $mutations</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; # A tibble: 52,274 x 23</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt;    chr     from     to ref   alt   gene  region cosmic function.</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;    </span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt;  1 chr1  8.36e7 8.36e7 C     T     LINC… inter… &lt;NA&gt;   &lt;NA&gt;     </span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt;  2 chr16 4.84e7 4.84e7 C     A     MIR5… ncRNA… &lt;NA&gt;   &lt;NA&gt;     </span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt;  3 chr16 4.88e7 4.88e7 G     T     MIR5… inter… &lt;NA&gt;   &lt;NA&gt;     </span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt;  4 chr16 4.89e7 4.89e7 T     G     MIR5… inter… &lt;NA&gt;   &lt;NA&gt;     </span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt;  5 chr16 4.92e7 4.92e7 A     C     MIR5… inter… &lt;NA&gt;   &lt;NA&gt;     </span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt;  6 chr16 4.92e7 4.92e7 T     C     MIR5… inter… &lt;NA&gt;   &lt;NA&gt;     </span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt;  7 chr16 4.92e7 4.92e7 A     G     MIR5… inter… &lt;NA&gt;   &lt;NA&gt;     </span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt;  8 chr16 4.92e7 4.92e7 G     A     MIR5… inter… &lt;NA&gt;   &lt;NA&gt;     </span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt;  9 chr16 4.92e7 4.92e7 C     G     MIR5… inter… &lt;NA&gt;   &lt;NA&gt;     </span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt; 10 chr16 4.92e7 4.92e7 A     T     MIR5… inter… &lt;NA&gt;   &lt;NA&gt;     </span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; # … with 52,264 more rows, and 14 more variables: mutlocation &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; #   patient &lt;chr&gt;, Set7_55.VAF &lt;dbl&gt;, Set7_57.VAF &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt; #   Set7_59.VAF &lt;dbl&gt;, Set7_62.VAF &lt;dbl&gt;, Set7_55.DP &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt; #   Set7_57.DP &lt;dbl&gt;, Set7_59.DP &lt;dbl&gt;, Set7_62.DP &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">#&gt; #   Set7_55.NV &lt;dbl&gt;, Set7_57.NV &lt;dbl&gt;, Set7_59.NV &lt;dbl&gt;, Set7_62.NV &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">#&gt; $CNA</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="co">#&gt; # A tibble: 60 x 11</span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="co">#&gt;    chr     from     to Set7_55.minor Set7_55.Major Set7_57.minor</span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt;         &lt;dbl&gt;         &lt;dbl&gt;         &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="co">#&gt;  1 chr1  1.58e6 2.26e8             1             1             1</span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="co">#&gt;  2 chr1  2.27e8 2.27e8             1             1             0</span></a>
<a class="sourceLine" id="cb3-28" data-line-number="28"><span class="co">#&gt;  3 chr1  2.27e8 2.48e8             1             1             0</span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="co">#&gt;  4 chr2  2.80e5 2.42e8             1             1             1</span></a>
<a class="sourceLine" id="cb3-30" data-line-number="30"><span class="co">#&gt;  5 chr3  3.86e5 1.41e8             1             1             1</span></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"><span class="co">#&gt;  6 chr3  1.41e8 1.81e8             1             2             1</span></a>
<a class="sourceLine" id="cb3-32" data-line-number="32"><span class="co">#&gt;  7 chr3  1.83e8 1.97e8             1             1             1</span></a>
<a class="sourceLine" id="cb3-33" data-line-number="33"><span class="co">#&gt;  8 chr4  7.20e5 6.32e6             1             1             1</span></a>
<a class="sourceLine" id="cb3-34" data-line-number="34"><span class="co">#&gt;  9 chr4  7.70e6 1.88e8             1             1             1</span></a>
<a class="sourceLine" id="cb3-35" data-line-number="35"><span class="co">#&gt; 10 chr5  1.71e6 6.51e7             1             1             1</span></a>
<a class="sourceLine" id="cb3-36" data-line-number="36"><span class="co">#&gt; # … with 50 more rows, and 5 more variables: Set7_57.Major &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb3-37" data-line-number="37"><span class="co">#&gt; #   Set7_59.minor &lt;dbl&gt;, Set7_59.Major &lt;dbl&gt;, Set7_62.minor &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb3-38" data-line-number="38"><span class="co">#&gt; #   Set7_62.Major &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb3-39" data-line-number="39"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-40" data-line-number="40"><span class="co">#&gt; $samples</span></a>
<a class="sourceLine" id="cb3-41" data-line-number="41"><span class="co">#&gt; [1] "Set7_55" "Set7_57" "Set7_59" "Set7_62"</span></a>
<a class="sourceLine" id="cb3-42" data-line-number="42"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-43" data-line-number="43"><span class="co">#&gt; $purity</span></a>
<a class="sourceLine" id="cb3-44" data-line-number="44"><span class="co">#&gt; Set7_55 Set7_57 Set7_59 Set7_62 </span></a>
<a class="sourceLine" id="cb3-45" data-line-number="45"><span class="co">#&gt;    0.88    0.88    0.88    0.80</span></a></code></pre></div>
</div>
<div id="analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#analysis" class="anchor"></a>Analysis</h3>
<p>First create a dataset from the <code>evoverse</code>, retaining only diploid chromosomes.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># Diploid segments </span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2">segments_karyo_diploid =<span class="st"> </span>MSEQ_CRC_ADENOCARCINOMA_SET7<span class="op">$</span>CNA <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>chr, <span class="op">-</span>from, <span class="op">-</span>to) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="dv">1</span>, <span class="cf">function</span>(x) <span class="kw"><a href="https://rdrr.io/r/base/all.html">all</a></span>(x <span class="op">==</span><span class="st"> </span><span class="dv">1</span>))</a>
<a class="sourceLine" id="cb4-5" data-line-number="5"></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(segments_karyo_diploid)</a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt;  [1]  TRUE FALSE FALSE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; [12]  TRUE  TRUE FALSE  TRUE FALSE  TRUE FALSE  TRUE  TRUE FALSE  TRUE</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt; [23] FALSE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE FALSE  TRUE FALSE FALSE</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt; [34] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt; [45] FALSE FALSE FALSE  TRUE FALSE  TRUE FALSE  TRUE  TRUE FALSE FALSE</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt; [56] FALSE FALSE FALSE FALSE FALSE</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co"># Actual dataset</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15">Set7 =<span class="st"> </span>evoverse<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/evoverse/man/dataset.html">dataset</a></span>(</a>
<a class="sourceLine" id="cb4-16" data-line-number="16">  <span class="dt">mutations =</span> MSEQ_CRC_ADENOCARCINOMA_SET7<span class="op">$</span>mutations, </a>
<a class="sourceLine" id="cb4-17" data-line-number="17">  <span class="dt">segments =</span> MSEQ_CRC_ADENOCARCINOMA_SET7<span class="op">$</span>CNA[segments_karyo_diploid, ],</a>
<a class="sourceLine" id="cb4-18" data-line-number="18">  <span class="dt">samples =</span> MSEQ_CRC_ADENOCARCINOMA_SET7<span class="op">$</span>samples,</a>
<a class="sourceLine" id="cb4-19" data-line-number="19">  <span class="dt">purity =</span> MSEQ_CRC_ADENOCARCINOMA_SET7<span class="op">$</span>purity,</a>
<a class="sourceLine" id="cb4-20" data-line-number="20">  <span class="dt">description =</span> <span class="st">"Set7 Sample from MSeq (multi-region WGS ~100x)"</span>  </a>
<a class="sourceLine" id="cb4-21" data-line-number="21">  )</a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="co">#&gt;  [ evoverse dataset ] </span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co">#&gt; Mutations N = 52274</span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="co">#&gt; Loading required package: CNAqc</span></a>
<a class="sourceLine" id="cb4-25" data-line-number="25"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-26" data-line-number="26"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-28" data-line-number="28"><span class="co">#&gt;  Creating CNAqc object for Set7_55</span></a>
<a class="sourceLine" id="cb4-29" data-line-number="29"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-30" data-line-number="30"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb4-31" data-line-number="31"><span class="co">#&gt; [CNAqc] Missing CCF column from CNA calls, adding CCF = 1 assuming all calls to be clonal.</span></a>
<a class="sourceLine" id="cb4-32" data-line-number="32"><span class="co">#&gt; Input  n = 52274 mutations for 25 CNA segments (25 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb4-33" data-line-number="33"><span class="co">#&gt; Mapped. n = 52274 mutations mapped to clonal segments (~100% of input) </span></a>
<a class="sourceLine" id="cb4-34" data-line-number="34"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-35" data-line-number="35"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-36" data-line-number="36"><span class="co">#&gt;  Creating CNAqc object for Set7_57</span></a>
<a class="sourceLine" id="cb4-37" data-line-number="37"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-38" data-line-number="38"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb4-39" data-line-number="39"><span class="co">#&gt; [CNAqc] Missing CCF column from CNA calls, adding CCF = 1 assuming all calls to be clonal.</span></a>
<a class="sourceLine" id="cb4-40" data-line-number="40"><span class="co">#&gt; Input  n = 52274 mutations for 25 CNA segments (25 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb4-41" data-line-number="41"><span class="co">#&gt; Mapped. n = 52274 mutations mapped to clonal segments (~100% of input) </span></a>
<a class="sourceLine" id="cb4-42" data-line-number="42"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-43" data-line-number="43"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-44" data-line-number="44"><span class="co">#&gt;  Creating CNAqc object for Set7_59</span></a>
<a class="sourceLine" id="cb4-45" data-line-number="45"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-46" data-line-number="46"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb4-47" data-line-number="47"><span class="co">#&gt; [CNAqc] Missing CCF column from CNA calls, adding CCF = 1 assuming all calls to be clonal.</span></a>
<a class="sourceLine" id="cb4-48" data-line-number="48"><span class="co">#&gt; Input  n = 52274 mutations for 25 CNA segments (25 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb4-49" data-line-number="49"><span class="co">#&gt; Mapped. n = 52274 mutations mapped to clonal segments (~100% of input) </span></a>
<a class="sourceLine" id="cb4-50" data-line-number="50"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-51" data-line-number="51"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-52" data-line-number="52"><span class="co">#&gt;  Creating CNAqc object for Set7_62</span></a>
<a class="sourceLine" id="cb4-53" data-line-number="53"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-54" data-line-number="54"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb4-55" data-line-number="55"><span class="co">#&gt; [CNAqc] Missing CCF column from CNA calls, adding CCF = 1 assuming all calls to be clonal.</span></a>
<a class="sourceLine" id="cb4-56" data-line-number="56"><span class="co">#&gt; Input  n = 52274 mutations for 25 CNA segments (25 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb4-57" data-line-number="57"><span class="co">#&gt; Mapped. n = 52274 mutations mapped to clonal segments (~100% of input) </span></a>
<a class="sourceLine" id="cb4-58" data-line-number="58"><span class="co">#&gt; Non-mappable mutations per chromosome 0</span></a>
<a class="sourceLine" id="cb4-59" data-line-number="59"></a>
<a class="sourceLine" id="cb4-60" data-line-number="60"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(Set7)</a>
<a class="sourceLine" id="cb4-61" data-line-number="61"><span class="co">#&gt;  [ mvMOBSTER dataset ] </span></a>
<a class="sourceLine" id="cb4-62" data-line-number="62"><span class="co">#&gt;   Dataset Set7 Sample from MSeq (multi-region WGS ~100x) </span></a>
<a class="sourceLine" id="cb4-63" data-line-number="63"><span class="co">#&gt;   Samples Set7_55, Set7_57, Set7_59, Set7_62 </span></a>
<a class="sourceLine" id="cb4-64" data-line-number="64"><span class="co">#&gt;  Purities 0.88, 0.88, 0.88, 0.8 </span></a>
<a class="sourceLine" id="cb4-65" data-line-number="65"><span class="co">#&gt; Mutations N = 52274 </span></a>
<a class="sourceLine" id="cb4-66" data-line-number="66"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-67" data-line-number="67"><span class="co">#&gt; MOBSTER analysis FALSE </span></a>
<a class="sourceLine" id="cb4-68" data-line-number="68"><span class="co">#&gt;   VIBER analysis FALSE </span></a>
<a class="sourceLine" id="cb4-69" data-line-number="69"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-70" data-line-number="70"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-71" data-line-number="71"><span class="co">#&gt;  LOGGED OPERATIONS</span></a>
<a class="sourceLine" id="cb4-72" data-line-number="72"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-73" data-line-number="73"><span class="co">#&gt; # A tibble: 1 x 2</span></a>
<a class="sourceLine" id="cb4-74" data-line-number="74"><span class="co">#&gt;   time                operation     </span></a>
<a class="sourceLine" id="cb4-75" data-line-number="75"><span class="co">#&gt;   &lt;dttm&gt;              &lt;chr&gt;         </span></a>
<a class="sourceLine" id="cb4-76" data-line-number="76"><span class="co">#&gt; 1 2019-11-12 13:46:19 Initialization</span></a></code></pre></div>
<p>Basic information can be accessed through the other functions of the <code>evoverse</code> packages. For instance, you can visualise CNA data through the <code>CNAqc</code> <a href="https://caravagn.github.op/CNAq">package</a>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># Example CNA rofiles with CNAqc</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(Set7<span class="op">$</span>CNAqc<span class="op">$</span>Set7_<span class="dv">55</span>)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt;       CNAqc  n = 52274 mutations for 25 CNA segments (25 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt;      Purity  88% cellularity </span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt;  Karyotypes  52274 (1:1) </span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt;  </span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt;    Peaks QC  FALSE </span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt;         CCF  FALSE</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co"># CNA segments plot</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11">ggpubr<span class="op">::</span><span class="kw"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span>(</a>
<a class="sourceLine" id="cb5-12" data-line-number="12">  <span class="dt">plotlist =</span> <span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(Set7<span class="op">$</span>CNAqc, <span class="cf">function</span>(x) { CNAqc<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/CNAqc/man/plot_segments.html">plot_segments</a></span>(x) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/graphics/plot.window.html">ylim</a></span>(<span class="dv">0</span>, <span class="dv">3</span>) }),</a>
<a class="sourceLine" id="cb5-13" data-line-number="13">  <span class="dt">ncol =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb5-14" data-line-number="14">  <span class="dt">nrow =</span> <span class="dv">4</span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15">)</a></code></pre></div>
<p><img src="Colorectal_Cross_files/figure-html/unnamed-chunk-5-1.png" width="960"></p>
<p>We can print also the VAF of the annotated mutations, in both 2D and 1D type of plots using <code>evoverse</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># plot by default is 2D</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">ggpubr<span class="op">::</span><span class="kw"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span>(</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">  <span class="dt">plotlist =</span> <span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(Set7),</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">  <span class="dt">ncol =</span> <span class="dv">3</span>, </a>
<a class="sourceLine" id="cb6-5" data-line-number="5">  <span class="dt">nrow =</span> <span class="dv">2</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6">)</a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; Plotting plain data - use clusters = {'MOBSTER', 'VIBER'} to plot clusters</span></a></code></pre></div>
<p><img src="Colorectal_Cross_files/figure-html/unnamed-chunk-6-1.png" width="864"></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># Using evoverse plot_1D_VAF (1D)</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2">ggpubr<span class="op">::</span><span class="kw"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span>(</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">  <span class="dt">plotlist =</span> <span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(Set7<span class="op">$</span>samples, evoverse<span class="op">::</span>plot_1D_VAF, <span class="dt">x =</span> Set7),</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">  <span class="dt">ncol =</span> <span class="dv">2</span>, </a>
<a class="sourceLine" id="cb7-5" data-line-number="5">  <span class="dt">nrow =</span> <span class="dv">2</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6">)</a></code></pre></div>
<p><img src="Colorectal_Cross_files/figure-html/unnamed-chunk-7-1.png" width="576"></p>
<p>The following steps can be run to analyse the cohort (here, they are not run). On a standard laptop this analysis takes about 10 minutes.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># MOBSTER analysis with default parameters</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2">Set7 =<span class="st">  </span>evoverse<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/evoverse/man/analyze_mobster.html">analyze_mobster</a></span>(Set7)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="co"># Save results</span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/r/base/save.html">save</a></span>(Set7, <span class="dt">file =</span> <span class="st">"MOBSTER_Set7_fits.RData"</span>)</a>
<a class="sourceLine" id="cb8-6" data-line-number="6"></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="co"># Plots of model selection that can be saved as PDF </span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8">figure =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(Set7<span class="op">$</span>fit_MOBSTER, mobster<span class="op">::</span>plot_model_selection) </a>
<a class="sourceLine" id="cb8-9" data-line-number="9">  </a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"MOBSTER_Set7_fits.pdf"</span>, <span class="dt">width =</span> <span class="dv">11</span>, <span class="dt">height =</span> <span class="dv">9</span>)</a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(figure, print)</a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a>
<a class="sourceLine" id="cb8-13" data-line-number="13"></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="co"># Remove tail mutations and replot the data</span></a>
<a class="sourceLine" id="cb8-15" data-line-number="15">Set7_notail =<span class="st"> </span><span class="kw"><a href="../reference/filter_tails.html">filter_tails</a></span>(Set7)</a>
<a class="sourceLine" id="cb8-16" data-line-number="16"></a>
<a class="sourceLine" id="cb8-17" data-line-number="17"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(Set7_notail)</a>
<a class="sourceLine" id="cb8-18" data-line-number="18"></a>
<a class="sourceLine" id="cb8-19" data-line-number="19"><span class="co"># VIBER clustering of Binomial mixtures for non-tail mutations</span></a>
<a class="sourceLine" id="cb8-20" data-line-number="20">Set7_notail =<span class="st"> </span>evoverse<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/evoverse/man/analyze_VIBER.html">analyze_VIBER</a></span>(Set7_notail, <span class="dt">K =</span> <span class="dv">7</span>, <span class="dt">alpha_0 =</span> <span class="fl">1e-06</span>, <span class="dt">max_iter =</span> <span class="dv">200</span>)</a>
<a class="sourceLine" id="cb8-21" data-line-number="21"></a>
<a class="sourceLine" id="cb8-22" data-line-number="22"><span class="kw"><a href="https://rdrr.io/r/base/save.html">save</a></span>(Set7_notail, <span class="dt">file =</span> <span class="st">'MOBSTER_VIBER_Set7_fits.RData'</span>)</a>
<a class="sourceLine" id="cb8-23" data-line-number="23">  </a>
<a class="sourceLine" id="cb8-24" data-line-number="24"><span class="co"># Plot </span></a>
<a class="sourceLine" id="cb8-25" data-line-number="25"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"MOBSTER_VIBER_Set7_fits_raw.pdf"</span>, <span class="dt">width =</span> <span class="dv">4</span>, <span class="dt">height =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb8-26" data-line-number="26"><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(Set7_notail, <span class="dt">clusters =</span> <span class="st">'VIBER'</span>), print)</a>
<a class="sourceLine" id="cb8-27" data-line-number="27"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a>
<a class="sourceLine" id="cb8-28" data-line-number="28">  </a>
<a class="sourceLine" id="cb8-29" data-line-number="29"><span class="co"># Apply heuristic (default parameters for instance) and replot</span></a>
<a class="sourceLine" id="cb8-30" data-line-number="30">reduced_mv =<span class="st"> </span>VIBER<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/VIBER/man/choose_clusters.html">choose_clusters</a></span>(Set7_notail<span class="op">$</span>fit_VIBER)</a>
<a class="sourceLine" id="cb8-31" data-line-number="31"></a>
<a class="sourceLine" id="cb8-32" data-line-number="32"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"MOBSTER_VIBER_Set7_fits_heuristics.pdf"</span>, <span class="dt">width =</span> <span class="dv">4</span>, <span class="dt">height =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb8-33" data-line-number="33"><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(VIBER<span class="op">:::</span><span class="kw">plot.vb_bmm</span>(reduced_mv), print)</a>
<a class="sourceLine" id="cb8-34" data-line-number="34"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
</div>
<div id="without-mobster" class="section level3">
<h3 class="hasAnchor">
<a href="#without-mobster" class="anchor"></a>Without MOBSTER</h3>
<p>For comparison, run this.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co"># VIBER clustering of Binomial mixtures for all mutations</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">Set7_standard =<span class="st"> </span>evoverse<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/evoverse/man/analyze_VIBER.html">analyze_VIBER</a></span>(Set7, <span class="dt">K =</span> <span class="dv">9</span>, <span class="dt">alpha_0 =</span> <span class="fl">1e-06</span>, <span class="dt">max_iter =</span> <span class="dv">200</span>)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/base/save.html">save</a></span>(Set7_standard, <span class="dt">file =</span> <span class="st">'VIBER_Set7_fits.RData'</span>)</a>
<a class="sourceLine" id="cb9-5" data-line-number="5">  </a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co"># Plot </span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"VIBER_Set7_fits_raw.pdf"</span>, <span class="dt">width =</span> <span class="dv">4</span>, <span class="dt">height =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(Set7_standard, <span class="dt">clusters =</span> <span class="st">'VIBER'</span>), print)</a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a>
<a class="sourceLine" id="cb9-10" data-line-number="10">  </a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co"># Apply heuristic (default parameters) and replot</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12">reduced_mv =<span class="st"> </span>VIBER<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/VIBER/man/choose_clusters.html">choose_clusters</a></span>(Set7_standard<span class="op">$</span>fit_VIBER)</a>
<a class="sourceLine" id="cb9-13" data-line-number="13"></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"VIBER_Set7_fits_heuristics.pdf"</span>, <span class="dt">width =</span> <span class="dv">4</span>, <span class="dt">height =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(VIBER<span class="op">:::</span><span class="kw">plot.vb_bmm</span>(reduced_mv), print)</a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.0.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
