<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of multi-sample datasets • mvmobster</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analysis of multi-sample datasets">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mvmobster</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/mvmobster.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Analysis.html">Cancer Evolution Analysis</a>
    </li>
    <li>
      <a href="../articles/Filtering.html">Query, filter, re-shape and plot datasets</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/mvMOBSTER">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Analysis of multi-sample datasets</h1>
                        <h4 class="author">Giulio Caravagna</h4>
            
            <h4 class="date">September 2019</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/mvMOBSTER/blob/master/vignettes/mvmobster.Rmd"><code>vignettes/mvmobster.Rmd</code></a></small>
      <div class="hidden name"><code>mvmobster.Rmd</code></div>

    </div>

    
    
<p>This simple vignette discusses how to prepare your data to be analyzed with this package.</p>
<p>See the Articles for vignettes about data query, modifications, analysis and plots.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">require</a></span>(tidyverse)</a></code></pre></div>
<div id="supported-input-formats" class="section level2">
<h2 class="hasAnchor">
<a href="#supported-input-formats" class="anchor"></a>Supported input formats</h2>
<p>A mvMOBSTER dataset contains, for <em>one or more</em> bulk samples</p>
<ul>
<li>
<strong>mutation data</strong>, in the usual format of substitutions involving one or more nucleotides;</li>
<li>
<strong>absolute copy number data</strong>, reported as segments profiles;</li>
<li>
<strong>sample purity</strong>, as the proportion of tumour cells in each one of the samples.</li>
</ul>
<p>For mutation and copy number data, two different types of data formats can be used - the <em>wide</em> and the <em>long</em> formats - and the package will determine automatically which one you are using.</p>
<p>All other parameters have a trivial data type, independently of how you want to pass mutation and copy number data.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Example input datasets are available just to show the possible</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="co"># formats in which data can be shaped before using mvMOBSTER</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(<span class="st">'example_input_formats'</span>, <span class="dt">package =</span> <span class="st">'mvmobster'</span>)</a></code></pre></div>
<blockquote>
<p>In the future, direct support of VCF input files will be implemented. Untill then you can use use packages like <code>vcfR</code> to extract mutation calls from VCF files</p>
</blockquote>
<div id="wide-format" class="section level4">
<h4 class="hasAnchor">
<a href="#wide-format" class="anchor"></a>Wide format</h4>
<p>This format is convenient if you generate joint calls across all samples, and you end up with a multi-sample VCF file, for instance.</p>
<p>The minimum information required are:</p>
<ul>
<li>A <code>dataframe</code> with mutation locations as <code>chr</code>, <code>from</code>, <code>to</code> in <code>hg19</code> coordinates, plus <code>ref</code> and <code>alt</code> for the reference and alternative alleles. For every sample <code>X</code>, this dataframe must contain columns <code>X.DP</code>, <code>X.NV</code> and <code>X.VAF</code> for the depth (total number of reads with both alleles), the number of reads with the alternative allele and the allele frequency so that <code>VAF = NV/DP</code>. Chromosome names must be in the format <code>chr1</code>, <code>chr2</code>, etc.; alleles should be characters and all other fields numeric.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># Example wide format for mutations - in this case columns region, gene and patient are extra</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">example_input_formats<span class="op">$</span>wide_mutations</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; # A tibble: 500 x 14</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt;    chr     from     to ref   alt   region gene  Set6_42.VAF Set6_42.DP</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;  &lt;int&gt;  &lt;int&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;       &lt;dbl&gt;      &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt;  1 chr5  8.98e7 8.98e7 G     A     inter… LYSM…       0              0</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt;  2 chrY  1.00e7 1.00e7 G     T     inter… TTTY…       0              0</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt;  3 chr12 7.29e7 7.29e7 C     A     intro… TRHDE       0              0</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt;  4 chr17 2.66e7 2.66e7 G     A     inter… NLK(…       0              0</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt;  5 chr11 1.14e7 1.14e7 T     G     intro… GALN…       0              0</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt;  6 chr1  5.47e7 5.47e7 A     G     intro… MRPL…       0              0</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt;  7 chr19 2.45e6 2.45e6 C     G     intro… LMNB2       0              0</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt;  8 chr10 8.48e7 8.48e7 A     C     inter… NRG3…       0              0</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt;  9 chr3  3.01e7 3.01e7 A     G     inter… RBMS…       0              0</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt; 10 chr6  1.00e7 1.00e7 C     T     inter… LOC1…       0.509         53</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; # … with 490 more rows, and 5 more variables: Set6_42.NV &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; #   Set6_44.VAF &lt;dbl&gt;, Set6_44.DP &lt;dbl&gt;, Set6_44.NV &lt;dbl&gt;, patient &lt;chr&gt;</span></a></code></pre></div>
<ul>
<li>A <code>dataframe</code> that contains copy number segments wkth <code>chr</code>, <code>from</code>, <code>to</code> in <code>hg19</code> coordinates, plus <code>X.Major</code> and <code>X.minor</code> for the number of copies of the major allele, and the minor (B-allele) allele in one of sample <code>X</code>’s segments. Chromosome names and other columns must be in the same format used by mutations.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># Example wide format for segments</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2">example_input_formats<span class="op">$</span>wide_segments</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt; # A tibble: 85 x 7</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt;    chr     from     to Set6_42.minor Set6_42.Major Set6_44.minor</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;  &lt;int&gt;  &lt;int&gt;         &lt;int&gt;         &lt;int&gt;         &lt;int&gt;</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt;  1 chr1  1.42e6 2.54e6             1             1             1</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt;  2 chr1  3.43e6 4.71e7             0             1             0</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt;  3 chr1  4.87e7 2.48e8             1             1             1</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt;  4 chr1  2.48e8 2.48e8             0             2             0</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt;  5 chr2  2.25e5 2.43e8             1             1             1</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt;  6 chr3  2.91e6 1.97e8             1             1             1</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt;  7 chr4  8.18e5 1.74e6             1             1             1</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt;  8 chr4  1.75e6 1.06e7             1             1             1</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt;  9 chr4  1.35e7 1.91e8             1             1             1</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt; 10 chr4  1.91e8 1.91e8             1             2             1</span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt; # … with 75 more rows, and 1 more variable: Set6_44.Major &lt;int&gt;</span></a></code></pre></div>
</div>
<div id="long-format" class="section level4">
<h4 class="hasAnchor">
<a href="#long-format" class="anchor"></a>Long format</h4>
<p>This format is convenient if you generate calls independently for all samples, and you end up with a VCF file per sample, for instance.</p>
<p>The minimum information required are the same of the wide format:</p>
<ul>
<li><p>A list of <code>dataframes</code>, where each dataframe contains mutations for a unique sample (instead than for all of the samples, as in the wide format). Mutation genomic coordinates must follow the same format as for the wide data, but this time the list entry for patient <code>X</code> can have columns <code>DP</code>, <code>NV</code> and <code>VAF</code> where the sample name does not appear. However, the list should be named so that the dataframe for sample <code>X</code> is named <code>X</code>.</p></li>
<li><p>A list of <code>dataframes</code>, where each dataframe contains segments. for a unique sample. Segments genomic coordinates must follow the same format as for the wide data, but this time the list entry for patient <code>X</code> can have columns <code>Major</code> and <code>minor</code> where the sample name does not appear. However, the list should be named so that the dataframe for sample <code>X</code> is named <code>X</code>.</p></li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># Example long format for mutations</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2">example_input_formats<span class="op">$</span>long_mutations</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; $Set42</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt; # A tibble: 175 x 11</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt;    chr      from     to ref   alt   region gene     VAF    DP    NV patient</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt;    &lt;chr&gt;   &lt;int&gt;  &lt;int&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;  </span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt;  1 chr6   1.00e7 1.00e7 C     T     inter… LOC1… 0.509     53    27 Set6   </span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt;  2 chr16  2.06e7 2.06e7 C     T     intro… ACSM… 0.174     46     8 Set6   </span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt;  3 chr3   1.80e8 1.80e8 G     A     inter… LOC1… 0.314     70    22 Set6   </span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt;  4 chr7   9.77e7 9.77e7 T     A     inter… OCM2… 0.0789    38     3 Set6   </span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt;  5 chr2   1.39e8 1.39e8 A     C     intro… NXPH2 0.0659    91     6 Set6   </span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt;  6 chr4   4.92e7 4.92e7 G     T     inter… CWH4… 0.167     48     8 Set6   </span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt;  7 chr20  5.30e7 5.30e7 C     T     inter… PFDN… 0.0721   111     8 Set6   </span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt;  8 chr15  4.39e7 4.39e7 C     G     intro… CATS… 0.0323    93     3 Set6   </span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt;  9 chr5   1.13e8 1.13e8 T     C     inter… YTHD… 0.479     94    45 Set6   </span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="co">#&gt; 10 chr19  4.07e6 4.07e6 G     T     upstr… ZBTB… 0.366     82    30 Set6   </span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="co">#&gt; # … with 165 more rows</span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="co">#&gt; $Set44</span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="co">#&gt; # A tibble: 146 x 11</span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="co">#&gt;    chr      from     to ref   alt   region gene     VAF    DP    NV patient</span></a>
<a class="sourceLine" id="cb5-22" data-line-number="22"><span class="co">#&gt;    &lt;chr&gt;   &lt;int&gt;  &lt;int&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;  </span></a>
<a class="sourceLine" id="cb5-23" data-line-number="23"><span class="co">#&gt;  1 chr12  7.29e7 7.29e7 C     A     intro… TRHDE 0.116     43     5 Set6   </span></a>
<a class="sourceLine" id="cb5-24" data-line-number="24"><span class="co">#&gt;  2 chr17  2.66e7 2.66e7 G     A     inter… NLK(… 0.127     63     8 Set6   </span></a>
<a class="sourceLine" id="cb5-25" data-line-number="25"><span class="co">#&gt;  3 chr6   1.00e7 1.00e7 C     T     inter… LOC1… 0.513     39    20 Set6   </span></a>
<a class="sourceLine" id="cb5-26" data-line-number="26"><span class="co">#&gt;  4 chr3   1.80e8 1.80e8 G     A     inter… LOC1… 0.358     67    24 Set6   </span></a>
<a class="sourceLine" id="cb5-27" data-line-number="27"><span class="co">#&gt;  5 chr7   9.77e7 9.77e7 T     A     inter… OCM2… 0.12      25     3 Set6   </span></a>
<a class="sourceLine" id="cb5-28" data-line-number="28"><span class="co">#&gt;  6 chr7   5.72e7 5.72e7 G     T     inter… ZNF4… 0.0702    57     4 Set6   </span></a>
<a class="sourceLine" id="cb5-29" data-line-number="29"><span class="co">#&gt;  7 chr9   1.33e8 1.33e8 T     G     intro… FNBP1 0.0794    63     5 Set6   </span></a>
<a class="sourceLine" id="cb5-30" data-line-number="30"><span class="co">#&gt;  8 chr5   1.13e8 1.13e8 T     C     inter… YTHD… 0.397     63    25 Set6   </span></a>
<a class="sourceLine" id="cb5-31" data-line-number="31"><span class="co">#&gt;  9 chrY   1.33e7 1.33e7 C     T     inter… NONE… 0.176     51     9 Set6   </span></a>
<a class="sourceLine" id="cb5-32" data-line-number="32"><span class="co">#&gt; 10 chr12  3.79e7 3.79e7 A     G     inter… NONE… 0.0541    74     4 Set6   </span></a>
<a class="sourceLine" id="cb5-33" data-line-number="33"><span class="co">#&gt; # … with 136 more rows</span></a></code></pre></div>
<ul>
<li>A <code>dataframe</code> that contains copy number segments wkth <code>chr</code>, <code>from</code>, <code>to</code> in <code>hg19</code> coordinates, plus <code>X.Major</code> and <code>X.minor</code> for the number of copies of the major allele, and the minor (B-allele) allele in each one of <code>X</code>’s segments.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># Example long format for segments</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">example_input_formats<span class="op">$</span>long_segments</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#&gt; $Set42</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt; # A tibble: 85 x 5</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt;    chr        from        to minor Major</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt;    &lt;chr&gt;     &lt;int&gt;     &lt;int&gt; &lt;int&gt; &lt;int&gt;</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt;  1 chr1    1421022   2538349     1     1</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt;  2 chr1    3428160  47139103     0     1</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt;  3 chr1   48708250 247694949     1     1</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt;  4 chr1  247726822 247726822     0     2</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt;  5 chr2     224919 242557019     1     1</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt;  6 chr3    2908335 197273077     1     1</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt;  7 chr4     818440   1739252     1     1</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt;  8 chr4    1746125  10599670     1     1</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt;  9 chr4   13529990 190894246     1     1</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt; 10 chr4  190903688 190904832     1     2</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt; # … with 75 more rows</span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#&gt; $Set44</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#&gt; # A tibble: 85 x 5</span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="co">#&gt;    chr        from        to minor Major</span></a>
<a class="sourceLine" id="cb6-22" data-line-number="22"><span class="co">#&gt;    &lt;chr&gt;     &lt;int&gt;     &lt;int&gt; &lt;int&gt; &lt;int&gt;</span></a>
<a class="sourceLine" id="cb6-23" data-line-number="23"><span class="co">#&gt;  1 chr1    1421022   2538349     1     1</span></a>
<a class="sourceLine" id="cb6-24" data-line-number="24"><span class="co">#&gt;  2 chr1    3428160  47139103     0     1</span></a>
<a class="sourceLine" id="cb6-25" data-line-number="25"><span class="co">#&gt;  3 chr1   48708250 247694949     1     1</span></a>
<a class="sourceLine" id="cb6-26" data-line-number="26"><span class="co">#&gt;  4 chr1  247726822 247726822     0     2</span></a>
<a class="sourceLine" id="cb6-27" data-line-number="27"><span class="co">#&gt;  5 chr2     224919 242557019     1     1</span></a>
<a class="sourceLine" id="cb6-28" data-line-number="28"><span class="co">#&gt;  6 chr3    2908335 197273077     1     1</span></a>
<a class="sourceLine" id="cb6-29" data-line-number="29"><span class="co">#&gt;  7 chr4     818440   1739252     1     1</span></a>
<a class="sourceLine" id="cb6-30" data-line-number="30"><span class="co">#&gt;  8 chr4    1746125  10599670     1     1</span></a>
<a class="sourceLine" id="cb6-31" data-line-number="31"><span class="co">#&gt;  9 chr4   13529990 190894246     1     1</span></a>
<a class="sourceLine" id="cb6-32" data-line-number="32"><span class="co">#&gt; 10 chr4  190903688 190904832     1     2</span></a>
<a class="sourceLine" id="cb6-33" data-line-number="33"><span class="co">#&gt; # … with 75 more rows</span></a></code></pre></div>
</div>
<div id="important-remarks" class="section level4">
<h4 class="hasAnchor">
<a href="#important-remarks" class="anchor"></a>Important remarks</h4>
<p>Data must be consistent, the package will check that raising warnings and error messages.</p>
<p>It is important that the user checks that:</p>
<ul>
<li>All mutations are annotated in every sample, <strong>even if absent</strong>. So even if a mutation has VAF 0 in a sample, its actual depth, NV, and VAF should be annotated anyway.</li>
<li>Copy number segments span the same intervals across all the input samples, i.e. like segments were joinly called, even if that is not the case.</li>
</ul>
<p>The first requirement is necessary to fit Binomial mixtures to the input data with the <a href="https://caravagn.github.io/VIBER/">VIBER</a> package for tumour subclonal deconvolution. VIBER implements variational fits for Binomial mixture models from read counts. The second requirement permits to easily switch between the wide and long representation of copy number data.</p>
<p><strong>Important note:</strong> this package uses the <a href="https://caravagn.github.io/CNAqc/">CNAqc</a> a package, which provide a set of metrics to assess the quality of Copy Number Alteration (CNA) calls. This package provides routines to map mutations to copy number segments, visualising copy number calls and assessing QC with quantitative metrics.</p>
</div>
</div>
<div id="creation-of-an-input-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#creation-of-an-input-dataset" class="anchor"></a>Creation of an input dataset</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(mvmobster)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#&gt;  [ CNAqc - Copy Number Alteration quality check ] </span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; GitHub :  caravagn/CNAqc </span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt;  [ MOBSTER - Model-based clustering in cancer ] </span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; GitHub :  caravagn/mobster </span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt;  &gt;  [https://github.com/caravagn/mvMOBSTER] See package "mvmobster" for support with multi-region sequencing analyses.</span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co">#&gt;  [ VIBER - Variational inference for Binomial mixtures ] </span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="co">#&gt; GitHub :  caravagn/VIBER</span></a></code></pre></div>
<p>With one of the associated formats.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">x =<span class="st"> </span><span class="kw"><a href="../reference/dataset.html">dataset</a></span>(</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">  <span class="dt">mutations =</span> example_input_formats<span class="op">$</span>long_mutations, </a>
<a class="sourceLine" id="cb8-3" data-line-number="3">  <span class="dt">segments =</span> example_input_formats<span class="op">$</span>long_segments, </a>
<a class="sourceLine" id="cb8-4" data-line-number="4">  <span class="dt">samples =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'Set42'</span>, <span class="st">'Set44'</span>), </a>
<a class="sourceLine" id="cb8-5" data-line-number="5">  <span class="dt">purity =</span>  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">`</span><span class="dt">Set42</span><span class="st">`</span> =<span class="st"> </span><span class="fl">0.8</span>, <span class="st">`</span><span class="dt">Set44</span><span class="st">`</span> =<span class="st"> </span><span class="fl">0.8</span>), </a>
<a class="sourceLine" id="cb8-6" data-line-number="6">  <span class="dt">description =</span> <span class="st">"A fake dataset"</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7">)</a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="co">#&gt;  [ mvMOBSTER dataset ]</span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="co">#&gt; Warning in dataset(mutations = example_input_formats$long_mutations,</span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="co">#&gt; segments = example_input_formats$long_segments, : NA values in the DP, NV</span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="co">#&gt; and VAF fields. Some mutations will be removed, you might want to check</span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="co">#&gt; your data....</span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="co">#&gt; Mutations N = 89</span></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="co">#&gt; Loading required package: CNAqc</span></a>
<a class="sourceLine" id="cb8-15" data-line-number="15"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb8-17" data-line-number="17"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb8-18" data-line-number="18"><span class="co">#&gt;  Creating CNAqc object for Set42</span></a>
<a class="sourceLine" id="cb8-19" data-line-number="19"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb8-20" data-line-number="20"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb8-21" data-line-number="21"><span class="co">#&gt; [CNAqc] Missing CCF column from CNA calls, adding CCF = 1 assuming all calls to be clonal.</span></a>
<a class="sourceLine" id="cb8-22" data-line-number="22"><span class="co">#&gt; Input  n = 89 mutations for 85 CNA segments (85 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb8-23" data-line-number="23"><span class="co">#&gt; Mapped. n = 81 mutations mapped to clonal segments (~91% of input) </span></a>
<a class="sourceLine" id="cb8-24" data-line-number="24"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb8-25" data-line-number="25"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb8-26" data-line-number="26"><span class="co">#&gt;  Creating CNAqc object for Set44</span></a>
<a class="sourceLine" id="cb8-27" data-line-number="27"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb8-28" data-line-number="28"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb8-29" data-line-number="29"><span class="co">#&gt; [CNAqc] Missing CCF column from CNA calls, adding CCF = 1 assuming all calls to be clonal.</span></a>
<a class="sourceLine" id="cb8-30" data-line-number="30"><span class="co">#&gt; Input  n = 89 mutations for 85 CNA segments (85 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb8-31" data-line-number="31"><span class="co">#&gt; Mapped. n = 81 mutations mapped to clonal segments (~91% of input) </span></a>
<a class="sourceLine" id="cb8-32" data-line-number="32"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb8-33" data-line-number="33"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb8-34" data-line-number="34"><span class="co">#&gt;  Non-mappable mutations</span></a>
<a class="sourceLine" id="cb8-35" data-line-number="35"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb8-36" data-line-number="36"><span class="co">#&gt; # A tibble: 16 x 4</span></a>
<a class="sourceLine" id="cb8-37" data-line-number="37"><span class="co">#&gt;    karyotype    VAF Set42                       Set44                      </span></a>
<a class="sourceLine" id="cb8-38" data-line-number="38"><span class="co">#&gt;    &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;                       &lt;chr&gt;                      </span></a>
<a class="sourceLine" id="cb8-39" data-line-number="39"><span class="co">#&gt;  1 &lt;NA&gt;      0.0667 chrX:151906569:151906569:A… &lt;NA&gt;                       </span></a>
<a class="sourceLine" id="cb8-40" data-line-number="40"><span class="co">#&gt;  2 &lt;NA&gt;      0.0789 &lt;NA&gt;                        chrX:151906569:151906569:A…</span></a>
<a class="sourceLine" id="cb8-41" data-line-number="41"><span class="co">#&gt;  3 &lt;NA&gt;      0.0944 chr15:21926248:21926248:G:T &lt;NA&gt;                       </span></a>
<a class="sourceLine" id="cb8-42" data-line-number="42"><span class="co">#&gt;  4 &lt;NA&gt;      0.121  &lt;NA&gt;                        chr15:21926248:21926248:G:T</span></a>
<a class="sourceLine" id="cb8-43" data-line-number="43"><span class="co">#&gt;  5 &lt;NA&gt;      0.136  &lt;NA&gt;                        chrY:22298291:22298291:A:G </span></a>
<a class="sourceLine" id="cb8-44" data-line-number="44"><span class="co">#&gt;  6 &lt;NA&gt;      0.175  &lt;NA&gt;                        chr21:10995460:10995460:C:T</span></a>
<a class="sourceLine" id="cb8-45" data-line-number="45"><span class="co">#&gt;  7 &lt;NA&gt;      0.205  chr21:10995460:10995460:C:T &lt;NA&gt;                       </span></a>
<a class="sourceLine" id="cb8-46" data-line-number="46"><span class="co">#&gt;  8 &lt;NA&gt;      0.239  chrY:22298291:22298291:A:G  &lt;NA&gt;                       </span></a>
<a class="sourceLine" id="cb8-47" data-line-number="47"><span class="co">#&gt;  9 &lt;NA&gt;      0.262  chr10:66993351:66993351:T:C &lt;NA&gt;                       </span></a>
<a class="sourceLine" id="cb8-48" data-line-number="48"><span class="co">#&gt; 10 &lt;NA&gt;      0.333  &lt;NA&gt;                        chr10:66993351:66993351:T:C</span></a>
<a class="sourceLine" id="cb8-49" data-line-number="49"><span class="co">#&gt; 11 &lt;NA&gt;      0.338  chr3:2189072:2189072:A:T    &lt;NA&gt;                       </span></a>
<a class="sourceLine" id="cb8-50" data-line-number="50"><span class="co">#&gt; 12 &lt;NA&gt;      0.348  chr19:23227155:23227155:C:T &lt;NA&gt;                       </span></a>
<a class="sourceLine" id="cb8-51" data-line-number="51"><span class="co">#&gt; 13 &lt;NA&gt;      0.413  &lt;NA&gt;                        chr19:23227155:23227155:C:T</span></a>
<a class="sourceLine" id="cb8-52" data-line-number="52"><span class="co">#&gt; 14 &lt;NA&gt;      0.492  &lt;NA&gt;                        chr3:2189072:2189072:A:T   </span></a>
<a class="sourceLine" id="cb8-53" data-line-number="53"><span class="co">#&gt; 15 &lt;NA&gt;      0.652  chrX:23146990:23146990:C:T  &lt;NA&gt;                       </span></a>
<a class="sourceLine" id="cb8-54" data-line-number="54"><span class="co">#&gt; 16 &lt;NA&gt;      0.704  &lt;NA&gt;                        chrX:23146990:23146990:C:T</span></a></code></pre></div>
<p><code>x</code> is an object of class <code>mbs_data</code>, which provides different S3 methods to print, visualize and query the loaded data.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co"># Display basic stats about the data</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(x)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="co">#&gt;  [ mvMOBSTER dataset ] </span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt;   Dataset A fake dataset </span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co">#&gt;   Samples Set42, Set44 </span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#&gt;  Purities 0.8, 0.8 </span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt; Mutations N = 81 </span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt; MOBSTER analysis FALSE </span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt;   VIBER analysis FALSE </span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt;  LOGGED OPERATIONS</span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="co">#&gt; # A tibble: 1 x 2</span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="co">#&gt;   time                operation     </span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#&gt;   &lt;dttm&gt;              &lt;chr&gt;         </span></a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="co">#&gt; 1 2019-09-23 00:44:00 Initialization</span></a>
<a class="sourceLine" id="cb9-19" data-line-number="19"></a>
<a class="sourceLine" id="cb9-20" data-line-number="20"><span class="co"># Summary of the measurements</span></a>
<a class="sourceLine" id="cb9-21" data-line-number="21"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/summary">summary</a></span>(x)</a>
<a class="sourceLine" id="cb9-22" data-line-number="22"><span class="co">#&gt;  [ mvMOBSTER dataset ] </span></a>
<a class="sourceLine" id="cb9-23" data-line-number="23"><span class="co">#&gt;   Dataset A fake dataset </span></a>
<a class="sourceLine" id="cb9-24" data-line-number="24"><span class="co">#&gt;   Samples Set42, Set44 </span></a>
<a class="sourceLine" id="cb9-25" data-line-number="25"><span class="co">#&gt;  Purities 0.8, 0.8 </span></a>
<a class="sourceLine" id="cb9-26" data-line-number="26"><span class="co">#&gt; Mutations N = 81 </span></a>
<a class="sourceLine" id="cb9-27" data-line-number="27"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-28" data-line-number="28"><span class="co">#&gt; MOBSTER analysis FALSE </span></a>
<a class="sourceLine" id="cb9-29" data-line-number="29"><span class="co">#&gt;   VIBER analysis FALSE </span></a>
<a class="sourceLine" id="cb9-30" data-line-number="30"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-31" data-line-number="31"><span class="co">#&gt; =-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb9-32" data-line-number="32"><span class="co">#&gt;  Mutation data</span></a>
<a class="sourceLine" id="cb9-33" data-line-number="33"><span class="co">#&gt; =-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb9-34" data-line-number="34"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-35" data-line-number="35"><span class="co">#&gt; Depth (DP)  </span></a>
<a class="sourceLine" id="cb9-36" data-line-number="36"><span class="co">#&gt; # A tibble: 2 x 6</span></a>
<a class="sourceLine" id="cb9-37" data-line-number="37"><span class="co">#&gt;   sample variable  mean median   min   max</span></a>
<a class="sourceLine" id="cb9-38" data-line-number="38"><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;    &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb9-39" data-line-number="39"><span class="co">#&gt; 1 Set42  DP        75.0     76    26   109</span></a>
<a class="sourceLine" id="cb9-40" data-line-number="40"><span class="co">#&gt; 2 Set44  DP        63.1     64    22   104</span></a>
<a class="sourceLine" id="cb9-41" data-line-number="41"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-42" data-line-number="42"><span class="co">#&gt; Number of reads with variant (NV)  </span></a>
<a class="sourceLine" id="cb9-43" data-line-number="43"><span class="co">#&gt; # A tibble: 2 x 6</span></a>
<a class="sourceLine" id="cb9-44" data-line-number="44"><span class="co">#&gt;   sample variable  mean median   min   max</span></a>
<a class="sourceLine" id="cb9-45" data-line-number="45"><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;    &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb9-46" data-line-number="46"><span class="co">#&gt; 1 Set42  NV        26.3     26     3    54</span></a>
<a class="sourceLine" id="cb9-47" data-line-number="47"><span class="co">#&gt; 2 Set44  NV        21.9     21     3    46</span></a>
<a class="sourceLine" id="cb9-48" data-line-number="48"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-49" data-line-number="49"><span class="co">#&gt; Variant allele frequency (VAF)  </span></a>
<a class="sourceLine" id="cb9-50" data-line-number="50"><span class="co">#&gt; # A tibble: 2 x 6</span></a>
<a class="sourceLine" id="cb9-51" data-line-number="51"><span class="co">#&gt;   sample variable  mean median   min   max</span></a>
<a class="sourceLine" id="cb9-52" data-line-number="52"><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;    &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb9-53" data-line-number="53"><span class="co">#&gt; 1 Set42  VAF       0.35   0.34  0.07  0.65</span></a>
<a class="sourceLine" id="cb9-54" data-line-number="54"><span class="co">#&gt; 2 Set44  VAF       0.34   0.34  0.05  0.65</span></a>
<a class="sourceLine" id="cb9-55" data-line-number="55"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-56" data-line-number="56"><span class="co">#&gt; Annotations region, gene, patient </span></a>
<a class="sourceLine" id="cb9-57" data-line-number="57"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-58" data-line-number="58"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb9-59" data-line-number="59"><span class="co">#&gt;  Copy Number Alteration data</span></a>
<a class="sourceLine" id="cb9-60" data-line-number="60"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb9-61" data-line-number="61"><span class="co">#&gt; $Set42</span></a>
<a class="sourceLine" id="cb9-62" data-line-number="62"><span class="co">#&gt;       CNAqc  n = 89 mutations for 85 CNA segments (85 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb9-63" data-line-number="63"><span class="co">#&gt;      Purity  80 cellularity </span></a>
<a class="sourceLine" id="cb9-64" data-line-number="64"><span class="co">#&gt;  Karyotypes  71 (1:1); 5 (2:0); 4 (1:0); 1 (2:1) </span></a>
<a class="sourceLine" id="cb9-65" data-line-number="65"><span class="co">#&gt;  </span></a>
<a class="sourceLine" id="cb9-66" data-line-number="66"><span class="co">#&gt;    Peaks QC  FALSE </span></a>
<a class="sourceLine" id="cb9-67" data-line-number="67"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-68" data-line-number="68"><span class="co">#&gt; $Set44</span></a>
<a class="sourceLine" id="cb9-69" data-line-number="69"><span class="co">#&gt;       CNAqc  n = 89 mutations for 85 CNA segments (85 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb9-70" data-line-number="70"><span class="co">#&gt;      Purity  80 cellularity </span></a>
<a class="sourceLine" id="cb9-71" data-line-number="71"><span class="co">#&gt;  Karyotypes  71 (1:1); 5 (2:0); 4 (1:0); 1 (2:1) </span></a>
<a class="sourceLine" id="cb9-72" data-line-number="72"><span class="co">#&gt;  </span></a>
<a class="sourceLine" id="cb9-73" data-line-number="73"><span class="co">#&gt;    Peaks QC  FALSE </span></a>
<a class="sourceLine" id="cb9-74" data-line-number="74"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-75" data-line-number="75"><span class="co">#&gt; [1] "stats about the fits...?"</span></a></code></pre></div>
<p>Regardless the input format, mutations are identified using as key the chromosomename, the genome location, the reference and alternative alleles.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(<span class="kw"><a href="../reference/keys.html">keys</a></span>(x))</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co">#&gt; [1] "chr6:10048079:10048079:C:T"   "chr3:180176259:180176259:G:A"</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="co">#&gt; [3] "chr7:97655791:97655791:T:A"   "chr5:113199069:113199069:T:C"</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#&gt; [5] "chr19:4066877:4066877:G:T"    "chr3:147617725:147617725:T:C"</span></a></code></pre></div>
<p>For instance, the SNV with a C&gt;T substitution at chromosome 6, hitting nucleotide 10048079 will have id <code>"chr6:10048079:10048079:C:T"</code>. The ids are used to create an internal indexed representation of the data, and the package provides several query functions to access data. Ids are accessible with function <code>keys</code>, and most query function can be subset with ids to retrieve specific mutations. Extra columns in the mutation and copy number data are retained in internal representation of the data, and won’t be manipulated.</p>
<p>The CNAqc objects generated when data have been loaded, are available inside <code>x</code>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(x<span class="op">$</span>CNAqc)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="co">#&gt; $Set42</span></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="co">#&gt;       CNAqc  n = 89 mutations for 85 CNA segments (85 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="co">#&gt;      Purity  80 cellularity </span></a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="co">#&gt;  Karyotypes  71 (1:1); 5 (2:0); 4 (1:0); 1 (2:1) </span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="co">#&gt;  </span></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="co">#&gt;    Peaks QC  FALSE </span></a>
<a class="sourceLine" id="cb11-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb11-9" data-line-number="9"><span class="co">#&gt; $Set44</span></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"><span class="co">#&gt;       CNAqc  n = 89 mutations for 85 CNA segments (85 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="co">#&gt;      Purity  80 cellularity </span></a>
<a class="sourceLine" id="cb11-12" data-line-number="12"><span class="co">#&gt;  Karyotypes  71 (1:1); 5 (2:0); 4 (1:0); 1 (2:1) </span></a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="co">#&gt;  </span></a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="co">#&gt;    Peaks QC  FALSE</span></a></code></pre></div>
<p>See the Articles for example of data query, modifications and plots.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#supported-input-formats">Supported input formats</a></li>
      <li><a href="#creation-of-an-input-dataset">Creation of an input dataset</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
