<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Pipeline to time aneuploidy. — pipeline_chromosome_timing • evoverse</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Pipeline to time aneuploidy. — pipeline_chromosome_timing" />
<meta property="og:description" content="For a list of input mutations, CNA and purity data from a tumour sample,
MOBSTER fits are computed to determine proportions of mutations in each
aneuploidy state (copy state). Assignments and plots are returned, and
an automatic QC pipeline (supervised logistic classifier) is used to
flag which fits pass quality check. The trained set for the classifier
consists in n = 377 colorectal whole genome samples." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">evoverse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/evoverse.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/0_packages.html">Packages inside `evoverse`</a>
    </li>
    <li>
      <a href="../articles/a1_Data_QC.html">P1. Data quality check</a>
    </li>
    <li>
      <a href="../articles/a2_Chr_timing.html">P2. Timing chromosome aneuploidy</a>
    </li>
    <li>
      <a href="../articles/a3_Deconv_vaf_karyo.html">P3. Subclonal deconvolution by VAF and karyotypes</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caravagn/evoverse">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Pipeline to time aneuploidy.</h1>
    <small class="dont-index">Source: <a href='https://github.com/caravagn/evoverse/blob/master/R/PIPELINE_CHR_TIMING.R'><code>R/PIPELINE_CHR_TIMING.R</code></a></small>
    <div class="hidden name"><code>pipeline_chromosome_timing.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>For a list of input mutations, CNA and purity data from a tumour sample,
MOBSTER fits are computed to determine proportions of mutations in each
aneuploidy state (copy state). Assignments and plots are returned, and
an automatic QC pipeline (supervised logistic classifier) is used to
flag which fits pass quality check. The trained set for the classifier
consists in n = 377 colorectal whole genome samples.</p>
    </div>

    <pre class="usage"><span class='fu'>pipeline_chromosome_timing</span>(<span class='no'>mutations</span>, <span class='no'>cna</span>, <span class='no'>purity</span>, <span class='kw'>karyotypes</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"2:0"</span>,
  <span class='st'>"2:1"</span>, <span class='st'>"2:2"</span>), <span class='kw'>reference</span> <span class='kw'>=</span> <span class='st'>"GRCh38"</span>, <span class='kw'>min_muts</span> <span class='kw'>=</span> <span class='fl'>150</span>,
  <span class='kw'>min_VAF</span> <span class='kw'>=</span> <span class='fl'>0.05</span>, <span class='kw'>description</span> <span class='kw'>=</span> <span class='st'>"Chromosomal timing sample"</span>,
  <span class='kw'>N_max</span> <span class='kw'>=</span> <span class='fl'>15000</span>, <span class='no'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>mutations</th>
      <td><p>Input mutations in the format for package <code>CNAqc</code>.</p></td>
    </tr>
    <tr>
      <th>cna</th>
      <td><p>Absolute Copy Number Alterations in the format for package <code>CNAqc</code>.</p></td>
    </tr>
    <tr>
      <th>purity</th>
      <td><p>Tumour purity, in [0, 1].</p></td>
    </tr>
    <tr>
      <th>karyotypes</th>
      <td><p>Karyotypes that can be timed, expressed as "Major:minor" notation. Default
are the following states <code>timeable = c('2:0', '2:1', '2:2')</code>.</p></td>
    </tr>
    <tr>
      <th>min_muts</th>
      <td><p>Skip analysing karyotypes with less than <code>min_muts</code> mutations.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Parameters passed <code>mobster_fit</code> in package <code>mobster</code>.</p></td>
    </tr>
    </table>


    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># We use data released with the CNAqc package</span>

<span class='no'>cna</span> <span class='kw'>=</span> <span class='kw pkg'>CNAqc</span><span class='kw ns'>::</span><span class='no'><a href='https://rdrr.io/pkg/CNAqc/man/example_dataset_CNAqc.html'>example_dataset_CNAqc</a></span>$<span class='no'>cna</span>
<span class='no'>mutations</span> <span class='kw'>=</span> <span class='kw pkg'>CNAqc</span><span class='kw ns'>::</span><span class='no'><a href='https://rdrr.io/pkg/CNAqc/man/example_dataset_CNAqc.html'>example_dataset_CNAqc</a></span>$<span class='no'>snvs</span>
<span class='no'>purity</span> <span class='kw'>=</span> <span class='kw pkg'>CNAqc</span><span class='kw ns'>::</span><span class='no'><a href='https://rdrr.io/pkg/CNAqc/man/example_dataset_CNAqc.html'>example_dataset_CNAqc</a></span>$<span class='no'>purity</span>

<span class='no'>x</span> <span class='kw'>=</span> <span class='fu'>pipeline_chromosome_timing</span>(<span class='no'>mutations</span>, <span class='kw'>cna</span> <span class='kw'>=</span> <span class='no'>cna</span>, <span class='kw'>purity</span> <span class='kw'>=</span> <span class='no'>purity</span>, <span class='kw'>auto_setup</span> <span class='kw'>=</span> <span class='st'>'FAST'</span>, <span class='kw'>N_max</span> <span class='kw'>=</span> <span class='fl'>500</span>)</div><div class='output co'>#&gt; <span style='color: #BBBBBB;background-color: #0000BB;'> [ Evoverse </span><span style='color: #BBBBBB;background-color: #0000BB;font-style: italic;'>Chromosomal timing pipeline</span><span style='color: #BBBBBB;background-color: #0000BB;'> ] 
#&gt; </span><span>
#&gt; </div><div class='output co'>#&gt; <span class='message'></span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Loading input data</span></div><div class='output co'>#&gt; 
#&gt; </div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>                    </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>──</span><span> </span><span style='font-weight: bold;'>Preparing input for deconvolution</span><span> </span><span style='color: #00BBBB;'>──────────────────────────────────────────────────────────────────</span><span></span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Loading input data</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>                    </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Removing 80 mutations with VAF &lt; 0.05.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Loading input data</span></div><div class='output co'>#&gt; </span><span style='color: #BBBBBB;background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'>                    </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Using reference genome coordinates for: </span><span style='color: #00BB00;'>GRCh38</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Loading input data</span></div><div class='output co'>#&gt; <span class='message'>                    </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Drivers are annotated, but 'gene' column is missing, using mutation location.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Loading input data</span></div><div class='output co'>#&gt; <span class='message'>                    </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Loading input data</span></div><div class='output co'>#&gt; <span class='message'>                    </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Input n = 13053 mutations for 267 CNA segments (267 clonal, 0 subclonal)</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Loading input data</span></div><div class='output co'>#&gt; <span class='message'>                    </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BB0000;'>x</span><span> n = 172 mutations  cannot be mapped to segments and will be removed.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Loading input data</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>                    </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> Mapped n = 12881 mutations to clonal segments (100% of input)</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Loading input data</span></div><div class='output co'>#&gt; <span class='message'>                    </span></div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #BBBB00;'>[ CNAqc ] </span><span> n = 12881 mutations in </span><span style='color: #00BB00;'>267</span><span> segments (267 clonal + 0 subclonal). Genome reference: </span><span style='color: #00BB00;'>GRCh38</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Loading input data</span></div><div class='output co'>#&gt; 
#&gt;    2:2  [n = 7437] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { </span><span style='color: #BBBB00;'>chr19:15292495:15292496:G:A</span><span> }
#&gt;    4:2  [n = 1873] ■■■■■■■
#&gt;    3:2  [n = 1612] ■■■■■■
#&gt;    2:1  [n = 1557] ■■■■■■  { </span><span style='color: #BBBB00;'>chr2:179431633:179431634:C:T</span><span> }
#&gt;    3:0  [n =  310] ■
#&gt; </div><div class='output co'>#&gt; <span class='message'>                    </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Sample Purity: 89% ~ Ploidy: 4.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Loading input data</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>                    </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Mutations annotated have 3 drivers.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Loading input data</span></div><div class='output co'>#&gt; <span class='message'>                    </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> Loading input data ... done</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; </div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>──</span><span> </span><span style='font-weight: bold;'>MOBSTER clustering mutations with karyotype </span><span style='color: #00BB00;font-weight: bold;'>2:1</span><span> </span><span style='color: #00BBBB;'>────────────────────────────────────────────────────</span><span></span></div><div class='output co'>#&gt; 
#&gt; ┌───────────────────────────────────────────────────────┐
#&gt; │                                                       │
#&gt; │   n = 1557 mutations. n &gt; 500 , downsampling input.   │
#&gt; │                                                       │
#&gt; └───────────────────────────────────────────────────────┘
#&gt; </span><span style='color: #BBBBBB;background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Using reference genome coordinates for: </span><span style='color: #00BB00;'>GRCh38</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Input n = 12881 mutations for 49 CNA segments (49 clonal, 0 subclonal)</span></div><div class='output co'>#&gt; <span class='warning'>Warning: [CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BB0000;'>x</span><span> n = 11324 mutations  cannot be mapped to segments and will be removed.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> Mapped n = 1557 mutations to clonal segments (100% of input)</span></div><div class='output co'>#&gt; </span><span style='color: #BBBBBB;background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Using reference genome coordinates for: </span><span style='color: #00BB00;'>GRCh38</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Input n = 500 mutations for 49 CNA segments (49 clonal, 0 subclonal)</span></div><div class='output co'>#&gt; <span class='warning'>Warning: [CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> Mapped n = 500 mutations to clonal segments (100% of input)</span></div><div class='output co'>#&gt; </span><span style='color: #BBBBBB;background-color: #0000BB;'> [ MOBSTER fit ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Loaded input data, n = 500.</span></div><div class='output co'>#&gt; <span class='message'>❯ n = 500. Mixture with k = </span><span style='color: #00BB00;'>1,2</span><span> Beta(s). Pareto tail: </span><span style='color: #00BB00;'>TRUE</span><span> and </span><span style='color: #00BB00;'>FALSE</span><span>. Output clusters with π &gt; 0.02</span>
#&gt; <span class='message'>and n &gt; 10.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> mobster automatic setup </span><span style='color: #00BB00;'>FAST</span><span> for the analysis.</span></div><div class='output co'>#&gt; <span class='message'>❯ Scoring (</span><span style='color: #BB0000;'>without parallel</span><span>) 2 x 2 x 2 = </span><span style='color: #00BB00;'>8</span><span> models by </span><span style='color: #00BB00;'>reICL</span><span>.</span></div><div class='output co'>#&gt; 
#&gt; |===============================================       | 88% ~0 s remaining     |======================================================|100% ~0 s remaining     
#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBBBB;'>ℹ</span><span> </span><span style='font-weight: bold;'>MOBSTER fits</span><span> completed in 3.4s.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #BBBB00;'>[ MOBSTER ] Raw VAF for 2:1</span><span> n = 500 with </span><span style='color: #00BB00;'>k = 1</span><span> Beta(s) </span><span style='color: #BB0000;'>without tail</span><span> ────────────────────────────────</span></div><div class='output co'>#&gt; <span class='message'>● Clusters: π = 100% [</span><span style='color: #BBBB00;'>C1</span><span>], with π &gt; 0.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BB0000;'>✖</span><span> No tail fit.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>● Beta </span><span style='color: #00BB00;'>C1</span><span> [n = 500, 100%] with mean = 0.43.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBBBB;'>ℹ</span><span> Score(s): NLL = -209.52; ICL = -400.39 (-400.39), H = 0 (0). Fit </span><span style='color: #BB0000;'>interrupted</span><span> by </span><span style='color: #00BB00;'>MM</span><span> in 100 steps.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Unknown or uninitialised column: 'Scale'.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Unknown or uninitialised column: 'Shape'.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Unknown or uninitialised column: 'Scale'.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Unknown or uninitialised column: 'Shape'.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BB0000;'>x</span><span> </span><span style='color: #BB0000;'>Raw VAF for 2:1</span><span>: QC FAIL. p = 0</span></div><div class='output co'>#&gt; 
#&gt; 
#&gt;         ┌──────────────────────────────────────────────────────────────────────────────────────┐
#&gt;         │</span><span style='background-color: #AF0000;'>                                                                                      </span><span>│
#&gt;         │</span><span style='background-color: #AF0000;'>   Auto-QC (type 'T') failed, attempting another single run with default parameters   </span><span>│
#&gt;         │</span><span style='background-color: #AF0000;'>                                                                                      </span><span>│
#&gt;         └──────────────────────────────────────────────────────────────────────────────────────┘
#&gt; </span><span style='color: #BBBBBB;background-color: #0000BB;'> [ MOBSTER fit ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Loaded input data, n = 500.</span></div><div class='output co'>#&gt; <span class='message'>❯ n = 500. Mixture with k = </span><span style='color: #00BB00;'>1,2,3</span><span> Beta(s). Pareto tail: </span><span style='color: #00BB00;'>TRUE</span><span> and </span><span style='color: #00BB00;'>FALSE</span><span>. Output clusters with π &gt; 0.02</span>
#&gt; <span class='message'>and n &gt; 10.</span></div><div class='output co'>#&gt; <span class='message'>❯ Custom fit by </span><span style='color: #00BB00;'>Moments-matching</span><span> in up to 250 steps, with ε = 1e-10 and </span><span style='color: #00BB00;'>peaks</span><span> initialisation.</span></div><div class='output co'>#&gt; <span class='message'>❯ Scoring (</span><span style='color: #BB0000;'>without parallel</span><span>) 5 x 3 x 2 = </span><span style='color: #00BB00;'>30</span><span> models by </span><span style='color: #00BB00;'>reICL</span><span>.</span></div><div class='output co'>#&gt; 
#&gt; |=======                                               | 13% ~22 s remaining    |=========                                             | 17% ~17 s remaining    |==========                                            | 20% ~15 s remaining    |============                                          | 23% ~20 s remaining    |==============                                        | 27% ~18 s remaining    |================                                      | 30% ~16 s remaining    |==================                                    | 33% ~19 s remaining    |===================                                   | 37% ~17 s remaining    |=====================                                 | 40% ~15 s remaining    |=======================                               | 43% ~16 s remaining    |=========================                             | 47% ~15 s remaining    |===========================                           | 50% ~14 s remaining    |============================                          | 53% ~14 s remaining    |==============================                        | 57% ~13 s remaining    |================================                      | 60% ~11 s remaining    |==================================                    | 63% ~11 s remaining    |====================================                  | 67% ~10 s remaining    |=====================================                 | 70% ~9 s remaining     |=======================================               | 73% ~9 s remaining     |=========================================             | 77% ~8 s remaining     |===========================================           | 80% ~6 s remaining     |=============================================         | 83% ~6 s remaining     |==============================================        | 87% ~5 s remaining     |================================================      | 90% ~3 s remaining     |==================================================    | 93% ~2 s remaining     |====================================================  | 97% ~1 s remaining     |======================================================|100% ~0 s remaining     
#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBBBB;'>ℹ</span><span> </span><span style='font-weight: bold;'>MOBSTER fits</span><span> completed in 36.2s.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #BBBB00;'>[ MOBSTER ] Raw VAF for 2:1</span><span> n = 500 with </span><span style='color: #00BB00;'>k = 2</span><span> Beta(s) </span><span style='color: #00BB00;'>and a tail</span><span> ──────────────────────────────────</span></div><div class='output co'>#&gt; <span class='message'>● Clusters: π = 52% [</span><span style='color: #BBBB00;'>C2</span><span>], 43% [</span><span style='color: #BBBB00;'>C1</span><span>] and 4% [</span><span style='color: #BBBB00;'>Tail</span><span>], with π &gt; 0.</span></div><div class='output co'>#&gt; <span class='message'>● Tail [n = 18, 4.3%] with alpha = 1.3.</span></div><div class='output co'>#&gt; <span class='message'>● Beta </span><span style='color: #00BB00;'>C1</span><span> [n = 217, 43%] with mean = 0.6.</span></div><div class='output co'>#&gt; <span class='message'>● Beta </span><span style='color: #00BB00;'>C2</span><span> [n = 265, 52%] with mean = 0.31.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBBBB;'>ℹ</span><span> Score(s): NLL = -416.88; ICL = -754.67 (-776.27), H = 23.17 (1.57). Fit </span><span style='color: #00BB00;'>converged</span><span> by </span><span style='color: #00BB00;'>MM</span><span> in 48 steps.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> </span><span style='color: #00BB00;'>Raw VAF for 2:1</span><span>: QC PASS. p = 0.992</span></div><div class='output co'>#&gt; 
#&gt; </div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>──</span><span> </span><span style='font-weight: bold;'>MOBSTER clustering mutations with karyotype </span><span style='color: #00BB00;font-weight: bold;'>2:2</span><span> </span><span style='color: #00BBBB;'>────────────────────────────────────────────────────</span><span></span></div><div class='output co'>#&gt; 
#&gt; ┌───────────────────────────────────────────────────────┐
#&gt; │                                                       │
#&gt; │   n = 7437 mutations. n &gt; 500 , downsampling input.   │
#&gt; │                                                       │
#&gt; └───────────────────────────────────────────────────────┘
#&gt; </span><span style='color: #BBBBBB;background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Using reference genome coordinates for: </span><span style='color: #00BB00;'>GRCh38</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Input n = 12881 mutations for 80 CNA segments (80 clonal, 0 subclonal)</span></div><div class='output co'>#&gt; <span class='warning'>Warning: [CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BB0000;'>x</span><span> n = 5444 mutations  cannot be mapped to segments and will be removed.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> Mapped n = 7437 mutations to clonal segments (100% of input)</span></div><div class='output co'>#&gt; </span><span style='color: #BBBBBB;background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Using reference genome coordinates for: </span><span style='color: #00BB00;'>GRCh38</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Input n = 500 mutations for 80 CNA segments (80 clonal, 0 subclonal)</span></div><div class='output co'>#&gt; <span class='warning'>Warning: [CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> Mapped n = 500 mutations to clonal segments (100% of input)</span></div><div class='output co'>#&gt; </span><span style='color: #BBBBBB;background-color: #0000BB;'> [ MOBSTER fit ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Loaded input data, n = 500.</span></div><div class='output co'>#&gt; <span class='message'>❯ n = 500. Mixture with k = </span><span style='color: #00BB00;'>1,2</span><span> Beta(s). Pareto tail: </span><span style='color: #00BB00;'>TRUE</span><span> and </span><span style='color: #00BB00;'>FALSE</span><span>. Output clusters with π &gt; 0.02</span>
#&gt; <span class='message'>and n &gt; 10.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> mobster automatic setup </span><span style='color: #00BB00;'>FAST</span><span> for the analysis.</span></div><div class='output co'>#&gt; <span class='message'>❯ Scoring (</span><span style='color: #BB0000;'>without parallel</span><span>) 2 x 2 x 2 = </span><span style='color: #00BB00;'>8</span><span> models by </span><span style='color: #00BB00;'>reICL</span><span>.</span></div><div class='output co'>#&gt; 
#&gt; |======================================================|100% ~0 s remaining     
#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBBBB;'>ℹ</span><span> </span><span style='font-weight: bold;'>MOBSTER fits</span><span> completed in 2.5s.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #BBBB00;'>[ MOBSTER ] Raw VAF for 2:2</span><span> n = 500 with </span><span style='color: #00BB00;'>k = 1</span><span> Beta(s) </span><span style='color: #00BB00;'>and a tail</span><span> ──────────────────────────────────</span></div><div class='output co'>#&gt; <span class='message'>● Clusters: π = 86% [</span><span style='color: #BBBB00;'>C1</span><span>] and 14% [</span><span style='color: #BBBB00;'>Tail</span><span>], with π &gt; 0.</span></div><div class='output co'>#&gt; <span class='message'>● Tail [n = 64, 14%] with alpha = 0.78.</span></div><div class='output co'>#&gt; <span class='message'>● Beta </span><span style='color: #00BB00;'>C1</span><span> [n = 436, 86%] with mean = 0.46.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBBBB;'>ℹ</span><span> Score(s): NLL = -611.19; ICL = -1157.82 (-1185.1), H = 27.28 (0). Fit </span><span style='color: #00BB00;'>converged</span><span> by </span><span style='color: #00BB00;'>MM</span><span> in 10 steps.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BB0000;'>x</span><span> </span><span style='color: #BB0000;'>Raw VAF for 2:2</span><span>: QC FAIL. p = 0</span></div><div class='output co'>#&gt; 
#&gt; 
#&gt;         ┌──────────────────────────────────────────────────────────────────────────────────────┐
#&gt;         │</span><span style='background-color: #AF0000;'>                                                                                      </span><span>│
#&gt;         │</span><span style='background-color: #AF0000;'>   Auto-QC (type 'T') failed, attempting another single run with default parameters   </span><span>│
#&gt;         │</span><span style='background-color: #AF0000;'>                                                                                      </span><span>│
#&gt;         └──────────────────────────────────────────────────────────────────────────────────────┘
#&gt; </span><span style='color: #BBBBBB;background-color: #0000BB;'> [ MOBSTER fit ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Loaded input data, n = 500.</span></div><div class='output co'>#&gt; <span class='message'>❯ n = 500. Mixture with k = </span><span style='color: #00BB00;'>1,2,3</span><span> Beta(s). Pareto tail: </span><span style='color: #00BB00;'>TRUE</span><span> and </span><span style='color: #00BB00;'>FALSE</span><span>. Output clusters with π &gt; 0.02</span>
#&gt; <span class='message'>and n &gt; 10.</span></div><div class='output co'>#&gt; <span class='message'>❯ Custom fit by </span><span style='color: #00BB00;'>Moments-matching</span><span> in up to 250 steps, with ε = 1e-10 and </span><span style='color: #00BB00;'>peaks</span><span> initialisation.</span></div><div class='output co'>#&gt; <span class='message'>❯ Scoring (</span><span style='color: #BB0000;'>without parallel</span><span>) 5 x 3 x 2 = </span><span style='color: #00BB00;'>30</span><span> models by </span><span style='color: #00BB00;'>reICL</span><span>.</span></div><div class='output co'>#&gt; 
#&gt; |=======                                               | 13% ~24 s remaining    |=========                                             | 17% ~19 s remaining    |==========                                            | 20% ~17 s remaining    |============                                          | 23% ~22 s remaining    |==============                                        | 27% ~19 s remaining    |================                                      | 30% ~20 s remaining    |==================                                    | 33% ~24 s remaining    |===================                                   | 37% ~20 s remaining    |=====================                                 | 40% ~18 s remaining    |=======================                               | 43% ~20 s remaining    |=========================                             | 47% ~17 s remaining    |===========================                           | 50% ~16 s remaining    |============================                          | 53% ~16 s remaining    |==============================                        | 57% ~14 s remaining    |================================                      | 60% ~13 s remaining    |==================================                    | 63% ~13 s remaining    |====================================                  | 67% ~11 s remaining    |=====================================                 | 70% ~10 s remaining    |=======================================               | 73% ~9 s remaining     |=========================================             | 77% ~8 s remaining     |===========================================           | 80% ~7 s remaining     |=============================================         | 83% ~6 s remaining     |==============================================        | 87% ~5 s remaining     |================================================      | 90% ~4 s remaining     |==================================================    | 93% ~3 s remaining     |====================================================  | 97% ~1 s remaining     |======================================================|100% ~0 s remaining     
#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBBBB;'>ℹ</span><span> </span><span style='font-weight: bold;'>MOBSTER fits</span><span> completed in 39.5s.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #BBBB00;'>[ MOBSTER ] Raw VAF for 2:2</span><span> n = 500 with </span><span style='color: #00BB00;'>k = 1</span><span> Beta(s) </span><span style='color: #00BB00;'>and a tail</span><span> ──────────────────────────────────</span></div><div class='output co'>#&gt; <span class='message'>● Clusters: π = 86% [</span><span style='color: #BBBB00;'>C1</span><span>] and 14% [</span><span style='color: #BBBB00;'>Tail</span><span>], with π &gt; 0.</span></div><div class='output co'>#&gt; <span class='message'>● Tail [n = 64, 14%] with alpha = 0.78.</span></div><div class='output co'>#&gt; <span class='message'>● Beta </span><span style='color: #00BB00;'>C1</span><span> [n = 436, 86%] with mean = 0.46.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBBBB;'>ℹ</span><span> Score(s): NLL = -611.19; ICL = -1157.82 (-1185.1), H = 27.28 (0). Fit </span><span style='color: #00BB00;'>converged</span><span> by </span><span style='color: #00BB00;'>MM</span><span> in 13 steps.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BB0000;'>x</span><span> </span><span style='color: #BB0000;'>Raw VAF for 2:2</span><span>: QC FAIL. p = 0</span></div><div class='output co'>#&gt; 
#&gt; </div><div class='output co'>#&gt; <span class='message'></span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Pipeline results assembly</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>                           </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> Pipeline results assembly ... done</span></div><div class='output co'>#&gt; </div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #BBBB00;'>[ Evoverse ] Chromosomal timing sample</span><span> Chromosome timing pipeline ──────────────────────────────────</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #BBBB00;'>[ CNAqc ] </span><span> n = 12881 mutations in </span><span style='color: #00BB00;'>267</span><span> segments (267 clonal + 0 subclonal). Genome reference: </span><span style='color: #00BB00;'>GRCh38</span><span>.</span></div><div class='output co'>#&gt; 
#&gt;    2:2  [n = 7437] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { </span><span style='color: #BBBB00;'>chr19:15292495:15292496:G:A</span><span> }
#&gt;    4:2  [n = 1873] ■■■■■■■
#&gt;    3:2  [n = 1612] ■■■■■■
#&gt;    2:1  [n = 1557] ■■■■■■  { </span><span style='color: #BBBB00;'>chr2:179431633:179431634:C:T</span><span> }
#&gt;    3:0  [n =  310] ■
#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Sample Purity: 89% ~ Ploidy: 4.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Mutations annotated have 3 drivers.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='background-color: #00BB00;'> QC PASS </span><span> ─────────────────────────────────────────────────────────────────────── PASS rate (%):  ──</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 1 x 4</span><span>
#&gt;   karyotype QC    QC_prob QC_type                
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                  
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 2:1       PASS    0.992 timing_logistic_GEL_CRC
#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='background-color: #BB0000;'> QC PASS </span><span> ─────────────────────────────────────────────────────────────────────── FAIL rate (%):  ──</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 1 x 4</span><span>
#&gt;   karyotype QC    QC_prob QC_type                
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                  
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 2:2       FAIL        0 timing_logistic_GEL_CRC
#&gt; 
#&gt;  </span><span style='background-color: #0000BB;'> LOG </span><span> 2020-04-26 02:56:52. evoverse pipeline for chromosome timing: with fits TRUE</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


