<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Summary for an <code>evoverse</code> dataset. — summary.mbst_data • evoverse</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Summary for an <code>evoverse</code> dataset. — summary.mbst_data" />
<meta property="og:description" content="The summary reports basic statistics on the stored
input values concerning both mutations, segments, and fits if
available" />
<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">evoverse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/evoverse.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/packages.html">Packages in the `evoverse`</a>
    </li>
    <li>
      <a href="../articles/pipelines.html">Pipeline available with the `evoverse` packages</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caravagn/evoverse">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Summary for an <code>evoverse</code> dataset.</h1>
    <small class="dont-index">Source: <a href='https://github.com/caravagn/evoverse/blob/master/R/S3_methods_print.R'><code>R/S3_methods_print.R</code></a></small>
    <div class="hidden name"><code>summary.mbst_data.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The summary reports basic statistics on the stored
input values concerning both mutations, segments, and fits if
available</p>
    </div>

    <pre class="usage"><span class='co'># S3 method for mbst_data</span>
<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span>(<span class='no'>x</span>, <span class='no'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>An  <code>evoverse</code> object.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>S3 parameters</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Nothing.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>'example_evoverse'</span>)
<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span>(<span class='no'>example_evoverse</span>)</div><div class='output co'>#&gt; <span style='color: #BBBBBB;background-color: #0000BB;'> [ mvMOBSTER dataset ] 
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>  Dataset</span><span> My multi-region MOBSTER dataset </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;   Samples</span><span> Set7_55, Set7_57, Set7_59, Set7_62 </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;  Purities</span><span> 0.88, 0.88, 0.88, 0.8 </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; Mutations</span><span> N = 57317 </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; </span><span>
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>MOBSTER analysis</span><span> </span><span style='color: #00BB00;'>TRUE</span><span> </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;   VIBER analysis</span><span> </span><span style='color: #BB0000;'>FALSE</span><span> </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; </span><span>
#&gt; </span><span style='color: #00BBBB;'>=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style='color: #00BBBB;'> Mutation data</span><span>
#&gt; </span><span style='color: #00BBBB;'>=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; Depth (DP)</span><span>  </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; </span><span style='color: #555555;'># A tibble: 4 x 6</span><span>
#&gt;   sample  variable  mean median   min   max
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'>1</span><span> Set7_55 DP        79.3     80     3  </span><span style='text-decoration: underline;'>3</span><span>577
#&gt; </span><span style='color: #555555;'>2</span><span> Set7_57 DP        77.7     78     3  </span><span style='text-decoration: underline;'>3</span><span>889
#&gt; </span><span style='color: #555555;'>3</span><span> Set7_59 DP        82.6     83     3  </span><span style='text-decoration: underline;'>3</span><span>823
#&gt; </span><span style='color: #555555;'>4</span><span> Set7_62 DP        90.3     90     6  </span><span style='text-decoration: underline;'>4</span><span>403
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; Number of reads with variant (NV)</span><span>  </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; </span><span style='color: #555555;'># A tibble: 4 x 6</span><span>
#&gt;   sample  variable  mean median   min   max
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'>1</span><span> Set7_55 NV        9.81      0     0   470
#&gt; </span><span style='color: #555555;'>2</span><span> Set7_57 NV        7.69      0     0   584
#&gt; </span><span style='color: #555555;'>3</span><span> Set7_59 NV        8.72      0     0   516
#&gt; </span><span style='color: #555555;'>4</span><span> Set7_62 NV        7.85      0     0   634
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; Variant allele frequency (VAF)</span><span>  </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; </span><span style='color: #555555;'># A tibble: 4 x 6</span><span>
#&gt;   sample  variable  mean median   min   max
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'>1</span><span> Set7_55 VAF       0.14      0     0  0.76
#&gt; </span><span style='color: #555555;'>2</span><span> Set7_57 VAF       0.11      0     0  0.76
#&gt; </span><span style='color: #555555;'>3</span><span> Set7_59 VAF       0.12      0     0  0.73
#&gt; </span><span style='color: #555555;'>4</span><span> Set7_62 VAF       0.11      0     0  0.76
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; Annotations</span><span> gene, vaf_normal, cosmic, function., mutlocation, region </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; </span><span>
#&gt; </span><span style='color: #00BBBB;'>=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style='color: #00BBBB;'> Copy Number Alteration data</span><span>
#&gt; </span><span style='color: #00BBBB;'>=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; $Set7_55
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>      CNAqc </span><span> n = 47357 mutations for 60 CNA segments (60 clonal, 0 subclonal) </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;      Purity </span><span> 88% cellularity </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;  Karyotypes </span><span> 47357 (1:1) 
#&gt;  </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;    Peaks QC </span><span> </span><span style='color: #BB0000;'>FALSE</span><span> </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;         CCF </span><span> </span><span style='color: #BB0000;'>FALSE</span><span> </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; </span><span>
#&gt; $Set7_57
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>      CNAqc </span><span> n = 47357 mutations for 60 CNA segments (60 clonal, 0 subclonal) </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;      Purity </span><span> 88% cellularity </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;  Karyotypes </span><span> 47357 (1:1) 
#&gt;  </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;    Peaks QC </span><span> </span><span style='color: #BB0000;'>FALSE</span><span> </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;         CCF </span><span> </span><span style='color: #BB0000;'>FALSE</span><span> </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; </span><span>
#&gt; $Set7_59
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>      CNAqc </span><span> n = 47357 mutations for 60 CNA segments (60 clonal, 0 subclonal) </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;      Purity </span><span> 88% cellularity </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;  Karyotypes </span><span> 47357 (1:1) 
#&gt;  </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;    Peaks QC </span><span> </span><span style='color: #BB0000;'>FALSE</span><span> </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;         CCF </span><span> </span><span style='color: #BB0000;'>FALSE</span><span> </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; </span><span>
#&gt; $Set7_62
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>      CNAqc </span><span> n = 47357 mutations for 60 CNA segments (60 clonal, 0 subclonal) </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;      Purity </span><span> 80% cellularity </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;  Karyotypes </span><span> 47357 (1:1) 
#&gt;  </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;    Peaks QC </span><span> </span><span style='color: #BB0000;'>FALSE</span><span> </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;         CCF </span><span> </span><span style='color: #BB0000;'>FALSE</span><span> </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; </span><span>
#&gt; 
#&gt; </span><span style='color: #00BBBB;'>=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style='color: #00BBBB;'> MOBSTER fit for sample Set7_55</span><span>
#&gt; </span><span style='color: #00BBBB;'>=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>[ MOBSTER ]</span><span> N = 20398 </span><span style='color: #00BBBB;'>points with</span><span> K = 1 </span><span style='color: #00BBBB;'>Beta clusters, fit by</span><span> </span><span style='color: #BBBB00;'>MM</span><span> </span><span style='color: #00BBBB;'>in</span><span> 10 </span><span style='color: #00BBBB;'>steps</span><span> </span><span style='color: #00BB00;'>[CONVERGED]</span><span> 
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;   Components (fit)  
#&gt; </span><span>     Tail	n = 9184 (0.47) 	 Shape = 1.7 
#&gt;   Beta C1 	n = 11214 (0.53) 	 Mean = 0.5
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;   Scores (model selection)  
#&gt; </span><span>       NLL       BIC       AIC  entropy       ICL reduced.entropy     reICL
#&gt;  -17664.43 -35269.33 -35316.87 1598.244 -33671.09               0 -35269.33
#&gt;  size
#&gt;     6
#&gt; 
#&gt; </span><span style='color: #00BBBB;'>=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style='color: #00BBBB;'> MOBSTER fit for sample Set7_57</span><span>
#&gt; </span><span style='color: #00BBBB;'>=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>[ MOBSTER ]</span><span> N = 19305 </span><span style='color: #00BBBB;'>points with</span><span> K = 1 </span><span style='color: #00BBBB;'>Beta clusters, fit by</span><span> </span><span style='color: #BBBB00;'>MM</span><span> </span><span style='color: #00BBBB;'>in</span><span> 10 </span><span style='color: #00BBBB;'>steps</span><span> </span><span style='color: #00BB00;'>[CONVERGED]</span><span> 
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;   Components (fit)  
#&gt; </span><span>     Tail	n = 11387 (0.61) 	 Shape = 1.4 
#&gt;   Beta C1 	n = 7918 (0.39) 	 Mean = 0.49
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;   Scores (model selection)  
#&gt; </span><span>       NLL       BIC       AIC entropy       ICL reduced.entropy     reICL size
#&gt;  -16489.14 -32919.08 -32966.29 2129.45 -30789.63               0 -32919.08    6
#&gt; 
#&gt; </span><span style='color: #00BBBB;'>=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style='color: #00BBBB;'> MOBSTER fit for sample Set7_59</span><span>
#&gt; </span><span style='color: #00BBBB;'>=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>[ MOBSTER ]</span><span> N = 22575 </span><span style='color: #00BBBB;'>points with</span><span> K = 1 </span><span style='color: #00BBBB;'>Beta clusters, fit by</span><span> </span><span style='color: #BBBB00;'>MM</span><span> </span><span style='color: #00BBBB;'>in</span><span> 9 </span><span style='color: #00BBBB;'>steps</span><span> </span><span style='color: #00BB00;'>[CONVERGED]</span><span> 
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;   Components (fit)  
#&gt; </span><span>     Tail	n = 14248 (0.65) 	 Shape = 1.8 
#&gt;   Beta C1 	n = 8327 (0.35) 	 Mean = 0.49
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;   Scores (model selection)  
#&gt; </span><span>       NLL       BIC      AIC  entropy       ICL reduced.entropy     reICL size
#&gt;  -24866.85 -49673.56 -49721.7 1641.651 -48031.91               0 -49673.56    6
#&gt; 
#&gt; </span><span style='color: #00BBBB;'>=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style='color: #00BBBB;'> MOBSTER fit for sample Set7_62</span><span>
#&gt; </span><span style='color: #00BBBB;'>=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>[ MOBSTER ]</span><span> N = 19852 </span><span style='color: #00BBBB;'>points with</span><span> K = 1 </span><span style='color: #00BBBB;'>Beta clusters, fit by</span><span> </span><span style='color: #BBBB00;'>MM</span><span> </span><span style='color: #00BBBB;'>in</span><span> 13 </span><span style='color: #00BBBB;'>steps</span><span> </span><span style='color: #00BB00;'>[CONVERGED]</span><span> 
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;   Components (fit)  
#&gt; </span><span>     Tail	n = 13106 (0.69) 	 Shape = 1.4 
#&gt;   Beta C1 	n = 6746 (0.31) 	 Mean = 0.5
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;   Scores (model selection)  
#&gt; </span><span>       NLL       BIC       AIC  entropy       ICL reduced.entropy     reICL
#&gt;  -17069.28 -34079.18 -34126.55 2326.259 -31752.92               0 -34079.18
#&gt;  size
#&gt;     6</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


